// ebay-api/lib/api/restful/developer/keyManagement/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const __1 = __importDefault(require("../../"));
/**
 * This method retrieves the call limit and utilization data for an application.
 */
class KeyManagement extends __1.default {
    get subdomain() {
        return 'apiz';
    }
    get basePath() {
        return '/developer/key_management/v1';
    }
    /**
     * This method returns the <b>Public Key</b>, <b>Public Key as JWE</b>,
     * and metadata for all keypairs associated with the application key making the call.
     */
    getSigningKeys() {
        return this.get(`/signing_key`);
    }
    /**
     * his method creates keypairs.
     */
    createSigningKey(data) {
        return this.post(`/signing_key`, data);
    }
    /**
     * This method returns the <b>Public Key</b>, <b>Public Key as JWE</b>,
     * and metadata for a specified <code>signingKeyId</code> associated with the application key making the call.
     * @param signingKeyId the signin key
     */
    getSigningKey(signingKeyId) {
        return this.get(`/signing_key/${signingKeyId}`);
    }
}
exports.default = KeyManagement;
KeyManagement.id = 'KeyManagement';
