// ebay-api/lib/api/traditional/fields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fields;
(function (fields) {
    fields[fields["AboutMePage"] = 0] = "AboutMePage";
    fields[fields["Accent"] = 1] = "Accent";
    fields[fields["AcceptedTime"] = 2] = "AcceptedTime";
    fields[fields["AccountCode"] = 3] = "AccountCode";
    fields[fields["AccountDetailsEntryType"] = 4] = "AccountDetailsEntryType";
    fields[fields["AccountEntries"] = 5] = "AccountEntries";
    fields[fields["AccountEntry"] = 6] = "AccountEntry";
    fields[fields["AccountEntrySortType"] = 7] = "AccountEntrySortType";
    fields[fields["AccountHistorySelection"] = 8] = "AccountHistorySelection";
    fields[fields["AccountID"] = 9] = "AccountID";
    fields[fields["AccountNumber"] = 10] = "AccountNumber";
    fields[fields["AccountState"] = 11] = "AccountState";
    fields[fields["AccountSummary"] = 12] = "AccountSummary";
    fields[fields["Ack"] = 13] = "Ack";
    fields[fields["Action"] = 14] = "Action";
    fields[fields["ActionRequiredBy"] = 15] = "ActionRequiredBy";
    fields[fields["ActiveAuctionCount"] = 16] = "ActiveAuctionCount";
    fields[fields["ActiveItemsOnly"] = 17] = "ActiveItemsOnly";
    fields[fields["ActiveList"] = 18] = "ActiveList";
    fields[fields["ActualDeliveryTime"] = 19] = "ActualDeliveryTime";
    fields[fields["ActualHandlingCost"] = 20] = "ActualHandlingCost";
    fields[fields["ActualShippingCost"] = 21] = "ActualShippingCost";
    fields[fields["AddItemRequestContainer"] = 22] = "AddItemRequestContainer";
    fields[fields["AddItemResponseContainer"] = 23] = "AddItemResponseContainer";
    fields[fields["AdditionalAccount"] = 24] = "AdditionalAccount";
    fields[fields["AdditionalCompatibilityEnabled"] = 25] = "AdditionalCompatibilityEnabled";
    fields[fields["AdditionalContactInformation"] = 26] = "AdditionalContactInformation";
    fields[fields["AdditionalInformation"] = 27] = "AdditionalInformation";
    fields[fields["AddMemberMessagesAAQToBidderRequestContainer"] = 28] = "AddMemberMessagesAAQToBidderRequestContainer";
    fields[fields["AddMemberMessagesAAQToBidderResponseContainer"] = 29] = "AddMemberMessagesAAQToBidderResponseContainer";
    fields[fields["Address"] = 30] = "Address";
    fields[fields["AddressID"] = 31] = "AddressID";
    fields[fields["AddressOwner"] = 32] = "AddressOwner";
    fields[fields["AddressUsage"] = 33] = "AddressUsage";
    fields[fields["AddToFavoriteStores"] = 34] = "AddToFavoriteStores";
    fields[fields["AdFormatEnabled"] = 35] = "AdFormatEnabled";
    fields[fields["AdFormatLead"] = 36] = "AdFormatLead";
    fields[fields["AdFormatLeadCount"] = 37] = "AdFormatLeadCount";
    fields[fields["AdjustmentAmount"] = 38] = "AdjustmentAmount";
    fields[fields["AdminEnded"] = 39] = "AdminEnded";
    fields[fields["AdminEndedItemsOnly"] = 40] = "AdminEndedItemsOnly";
    fields[fields["Adult"] = 41] = "Adult";
    fields[fields["AdvancedThemeArray"] = 42] = "AdvancedThemeArray";
    fields[fields["AffiliateTrackingDetails"] = 43] = "AffiliateTrackingDetails";
    fields[fields["AffiliateUserID"] = 44] = "AffiliateUserID";
    fields[fields["AffiliationType"] = 45] = "AffiliationType";
    fields[fields["Alert"] = 46] = "Alert";
    fields[fields["AlertEmail"] = 47] = "AlertEmail";
    fields[fields["AlertEnable"] = 48] = "AlertEnable";
    fields[fields["AlertType"] = 49] = "AlertType";
    fields[fields["AllAuctionItems"] = 50] = "AllAuctionItems";
    fields[fields["AllFeaturesForCategory"] = 51] = "AllFeaturesForCategory";
    fields[fields["AllFixedPriceItems"] = 52] = "AllFixedPriceItems";
    fields[fields["AllowPaymentEdit"] = 53] = "AllowPaymentEdit";
    fields[fields["AllStoreInventoryItems"] = 54] = "AllStoreInventoryItems";
    fields[fields["AlternateText"] = 55] = "AlternateText";
    fields[fields["AlwaysUseThisPaymentAddress"] = 56] = "AlwaysUseThisPaymentAddress";
    fields[fields["Amount"] = 57] = "Amount";
    fields[fields["AmountLimitRemaining"] = 58] = "AmountLimitRemaining";
    fields[fields["AmountPaid"] = 59] = "AmountPaid";
    fields[fields["AmountPastDue"] = 60] = "AmountPastDue";
    fields[fields["AmountSaved"] = 61] = "AmountSaved";
    fields[fields["AnnouncementStartTime"] = 62] = "AnnouncementStartTime";
    fields[fields["Answer1"] = 63] = "Answer1";
    fields[fields["Answer2"] = 64] = "Answer2";
    fields[fields["ApiAccessRule"] = 65] = "ApiAccessRule";
    fields[fields["ApplicationData"] = 66] = "ApplicationData";
    fields[fields["ApplicationDeliveryPreferences"] = 67] = "ApplicationDeliveryPreferences";
    fields[fields["ApplicationDeviceType"] = 68] = "ApplicationDeviceType";
    fields[fields["ApplicationEnable"] = 69] = "ApplicationEnable";
    fields[fields["ApplicationURL"] = 70] = "ApplicationURL";
    fields[fields["ApplyBuyerProtection"] = 71] = "ApplyBuyerProtection";
    fields[fields["Archived"] = 72] = "Archived";
    fields[fields["ASQPreferences"] = 73] = "ASQPreferences";
    fields[fields["Attribute"] = 74] = "Attribute";
    fields[fields["AttributeArray"] = 75] = "AttributeArray";
    fields[fields["AttributeRecommendations"] = 76] = "AttributeRecommendations";
    fields[fields["AttributeSetArray"] = 77] = "AttributeSetArray";
    fields[fields["AttributeSetID"] = 78] = "AttributeSetID";
    fields[fields["AttributeSetVersion"] = 79] = "AttributeSetVersion";
    fields[fields["AttributeSystemVersion"] = 80] = "AttributeSystemVersion";
    fields[fields["AuctionBidCount"] = 81] = "AuctionBidCount";
    fields[fields["AuctionSellingCount"] = 82] = "AuctionSellingCount";
    fields[fields["AudioChallengeURL"] = 83] = "AudioChallengeURL";
    fields[fields["AutomatedLeaveFeedbackEvent"] = 84] = "AutomatedLeaveFeedbackEvent";
    fields[fields["AutomatedListingRule"] = 85] = "AutomatedListingRule";
    fields[fields["AutomatedRelistingRule"] = 86] = "AutomatedRelistingRule";
    fields[fields["AutomatedSecondChanceOfferRule"] = 87] = "AutomatedSecondChanceOfferRule";
    fields[fields["AutomationAlert"] = 88] = "AutomationAlert";
    fields[fields["AutoOptDonationRefund"] = 89] = "AutoOptDonationRefund";
    fields[fields["AutoPay"] = 90] = "AutoPay";
    fields[fields["AutoPayEnabled"] = 91] = "AutoPayEnabled";
    fields[fields["AutoRelist"] = 92] = "AutoRelist";
    fields[fields["AverageRatingDetails"] = 93] = "AverageRatingDetails";
    fields[fields["AverageRatingSummary"] = 94] = "AverageRatingSummary";
    fields[fields["AverageSellingPrice"] = 95] = "AverageSellingPrice";
    fields[fields["AverageSoldPrice"] = 96] = "AverageSoldPrice";
    fields[fields["AverageStartPrice"] = 97] = "AverageStartPrice";
    fields[fields["B2BVATEnabled"] = 98] = "B2BVATEnabled";
    fields[fields["Balance"] = 99] = "Balance";
    fields[fields["BankAccountInfo"] = 100] = "BankAccountInfo";
    fields[fields["BankModifyDate"] = 101] = "BankModifyDate";
    fields[fields["BaseURL"] = 102] = "BaseURL";
    fields[fields["BasicThemeArray"] = 103] = "BasicThemeArray";
    fields[fields["BasicUpgradePackEnabled"] = 104] = "BasicUpgradePackEnabled";
    fields[fields["BeginDate"] = 105] = "BeginDate";
    fields[fields["BestOffer"] = 106] = "BestOffer";
    fields[fields["BestOfferArray"] = 107] = "BestOfferArray";
    fields[fields["BestOfferAutoAcceptEnabled"] = 108] = "BestOfferAutoAcceptEnabled";
    fields[fields["BestOfferAutoAcceptPrice"] = 109] = "BestOfferAutoAcceptPrice";
    fields[fields["BestOfferAutoDeclineEnabled"] = 110] = "BestOfferAutoDeclineEnabled";
    fields[fields["BestOfferCodeType"] = 111] = "BestOfferCodeType";
    fields[fields["BestOfferCount"] = 112] = "BestOfferCount";
    fields[fields["BestOfferCounterEnabled"] = 113] = "BestOfferCounterEnabled";
    fields[fields["BestOfferDetails"] = 114] = "BestOfferDetails";
    fields[fields["BestOfferEnabled"] = 115] = "BestOfferEnabled";
    fields[fields["BestOfferID"] = 116] = "BestOfferID";
    fields[fields["BestOfferList"] = 117] = "BestOfferList";
    fields[fields["BestOfferSale"] = 118] = "BestOfferSale";
    fields[fields["BestOfferStatus"] = 119] = "BestOfferStatus";
    fields[fields["BestTimeToCall"] = 120] = "BestTimeToCall";
    fields[fields["BidActivityWithSeller"] = 121] = "BidActivityWithSeller";
    fields[fields["BidArray"] = 122] = "BidArray";
    fields[fields["BidAssistantList"] = 123] = "BidAssistantList";
    fields[fields["BidCount"] = 124] = "BidCount";
    fields[fields["BidCountMax"] = 125] = "BidCountMax";
    fields[fields["BidCountMin"] = 126] = "BidCountMin";
    fields[fields["Bidder"] = 127] = "Bidder";
    fields[fields["BidderCount"] = 128] = "BidderCount";
    fields[fields["BidderNoticePreferences"] = 129] = "BidderNoticePreferences";
    fields[fields["BiddingCount"] = 130] = "BiddingCount";
    fields[fields["BiddingDetails"] = 131] = "BiddingDetails";
    fields[fields["BiddingSummary"] = 132] = "BiddingSummary";
    fields[fields["BidIncrement"] = 133] = "BidIncrement";
    fields[fields["BidItemArray"] = 134] = "BidItemArray";
    fields[fields["BidList"] = 135] = "BidList";
    fields[fields["BidRetractionFeedbackPeriodArray"] = 136] = "BidRetractionFeedbackPeriodArray";
    fields[fields["BidRetractions"] = 137] = "BidRetractions";
    fields[fields["BidsToUniqueCategories"] = 138] = "BidsToUniqueCategories";
    fields[fields["BidsToUniqueSellers"] = 139] = "BidsToUniqueSellers";
    fields[fields["BillingCycleDate"] = 140] = "BillingCycleDate";
    fields[fields["BillingEmail"] = 141] = "BillingEmail";
    fields[fields["BindingAuction"] = 142] = "BindingAuction";
    fields[fields["BlockOnWarning"] = 143] = "BlockOnWarning";
    fields[fields["Body"] = 144] = "Body";
    fields[fields["BoldTitle"] = 145] = "BoldTitle";
    fields[fields["Border"] = 146] = "Border";
    fields[fields["BotBlock"] = 147] = "BotBlock";
    fields[fields["BotBlockAudioUrl"] = 148] = "BotBlockAudioUrl";
    fields[fields["BotBlockToken"] = 149] = "BotBlockToken";
    fields[fields["BotBlockUrl"] = 150] = "BotBlockUrl";
    fields[fields["BotBlockUserInput"] = 151] = "BotBlockUserInput";
    fields[fields["Brand"] = 152] = "Brand";
    fields[fields["BrandMPN"] = 153] = "BrandMPN";
    fields[fields["BrandMPNIdentifierEnabled"] = 154] = "BrandMPNIdentifierEnabled";
    fields[fields["BriefText"] = 155] = "BriefText";
    fields[fields["Build"] = 156] = "Build";
    fields[fields["BusinessRole"] = 157] = "BusinessRole";
    fields[fields["BusinessSeller"] = 158] = "BusinessSeller";
    fields[fields["BusinessSellerDetails"] = 159] = "BusinessSellerDetails";
    fields[fields["Buyer"] = 160] = "Buyer";
    fields[fields["BuyerCheckoutMessage"] = 161] = "BuyerCheckoutMessage";
    fields[fields["BuyerEmail"] = 162] = "BuyerEmail";
    fields[fields["BuyerExternalTransactionID"] = 163] = "BuyerExternalTransactionID";
    fields[fields["BuyerGuaranteeEnabled"] = 164] = "BuyerGuaranteeEnabled";
    fields[fields["BuyerGuaranteePrice"] = 165] = "BuyerGuaranteePrice";
    fields[fields["BuyerID"] = 166] = "BuyerID";
    fields[fields["BuyerInfo"] = 167] = "BuyerInfo";
    fields[fields["BuyerMessage"] = 168] = "BuyerMessage";
    fields[fields["BuyerPaidStatus"] = 169] = "BuyerPaidStatus";
    fields[fields["BuyerProtection"] = 170] = "BuyerProtection";
    fields[fields["BuyerProtectionSource"] = 171] = "BuyerProtectionSource";
    fields[fields["BuyerProtectionStatus"] = 172] = "BuyerProtectionStatus";
    fields[fields["BuyerRequirementDetails"] = 173] = "BuyerRequirementDetails";
    fields[fields["BuyerResponsibleForShipping"] = 174] = "BuyerResponsibleForShipping";
    fields[fields["BuyerRoleMetrics"] = 175] = "BuyerRoleMetrics";
    fields[fields["BuyerSatisfaction"] = 176] = "BuyerSatisfaction";
    fields[fields["BuyerSelectedShipping"] = 177] = "BuyerSelectedShipping";
    fields[fields["BuyerUserID"] = 178] = "BuyerUserID";
    fields[fields["BuyingReminders"] = 179] = "BuyingReminders";
    fields[fields["BuyingSummary"] = 180] = "BuyingSummary";
    fields[fields["BuyItNowAdded"] = 181] = "BuyItNowAdded";
    fields[fields["BuyItNowAvailable"] = 182] = "BuyItNowAvailable";
    fields[fields["BuyItNowLowered"] = 183] = "BuyItNowLowered";
    fields[fields["BuyItNowPrice"] = 184] = "BuyItNowPrice";
    fields[fields["CalculatedHandlingDiscount"] = 185] = "CalculatedHandlingDiscount";
    fields[fields["CalculatedShippingDiscount"] = 186] = "CalculatedShippingDiscount";
    fields[fields["CalculatedShippingRate"] = 187] = "CalculatedShippingRate";
    fields[fields["CallMode"] = 188] = "CallMode";
    fields[fields["CallName"] = 189] = "CallName";
    fields[fields["CallStatus"] = 190] = "CallStatus";
    fields[fields["CampaignDisplayName"] = 191] = "CampaignDisplayName";
    fields[fields["CampaignID"] = 192] = "CampaignID";
    fields[fields["CancelCompleteDate"] = 193] = "CancelCompleteDate";
    fields[fields["CancelDetail"] = 194] = "CancelDetail";
    fields[fields["CancelIntiationDate"] = 195] = "CancelIntiationDate";
    fields[fields["CancelIntiator"] = 196] = "CancelIntiator";
    fields[fields["CancelReason"] = 197] = "CancelReason";
    fields[fields["CancelReasonDetails"] = 198] = "CancelReasonDetails";
    fields[fields["CancelStatus"] = 199] = "CancelStatus";
    fields[fields["CarrierID"] = 200] = "CarrierID";
    fields[fields["CartID"] = 201] = "CartID";
    fields[fields["CashOnDeliveryCost"] = 202] = "CashOnDeliveryCost";
    fields[fields["CatalogEnabled"] = 203] = "CatalogEnabled";
    fields[fields["Category"] = 204] = "Category";
    fields[fields["Category2ID"] = 205] = "Category2ID";
    fields[fields["CategoryArray"] = 206] = "CategoryArray";
    fields[fields["CategoryBasedAttributesPrefill"] = 207] = "CategoryBasedAttributesPrefill";
    fields[fields["CategoryCount"] = 208] = "CategoryCount";
    fields[fields["CategoryGroup"] = 209] = "CategoryGroup";
    fields[fields["CategoryID"] = 210] = "CategoryID";
    fields[fields["CategoryLevel"] = 211] = "CategoryLevel";
    fields[fields["CategoryMapping"] = 212] = "CategoryMapping";
    fields[fields["CategoryMappingAllowed"] = 213] = "CategoryMappingAllowed";
    fields[fields["CategoryName"] = 214] = "CategoryName";
    fields[fields["CategoryParent"] = 215] = "CategoryParent";
    fields[fields["CategoryParentID"] = 216] = "CategoryParentID";
    fields[fields["CategoryParentName"] = 217] = "CategoryParentName";
    fields[fields["CategorySiteID"] = 218] = "CategorySiteID";
    fields[fields["CategorySpecific"] = 219] = "CategorySpecific";
    fields[fields["CategorySpecificsFileInfo"] = 220] = "CategorySpecificsFileInfo";
    fields[fields["CategoryStructureOnly"] = 221] = "CategoryStructureOnly";
    fields[fields["CategoryVersion"] = 222] = "CategoryVersion";
    fields[fields["CeilingPrice"] = 223] = "CeilingPrice";
    fields[fields["ChallengeToken"] = 224] = "ChallengeToken";
    fields[fields["ChangePaymentInstructions"] = 225] = "ChangePaymentInstructions";
    fields[fields["CharacteristicsSet"] = 226] = "CharacteristicsSet";
    fields[fields["CharacteristicsSets"] = 227] = "CharacteristicsSets";
    fields[fields["Charity"] = 228] = "Charity";
    fields[fields["CharityAffiliationDetail"] = 229] = "CharityAffiliationDetail";
    fields[fields["CharityAffiliationDetails"] = 230] = "CharityAffiliationDetails";
    fields[fields["CharityAffiliations"] = 231] = "CharityAffiliations";
    fields[fields["CharityDomain"] = 232] = "CharityDomain";
    fields[fields["CharityID"] = 233] = "CharityID";
    fields[fields["CharityListing"] = 234] = "CharityListing";
    fields[fields["CharityName"] = 235] = "CharityName";
    fields[fields["CharityNumber"] = 236] = "CharityNumber";
    fields[fields["CharityRegion"] = 237] = "CharityRegion";
    fields[fields["CharityRegistered"] = 238] = "CharityRegistered";
    fields[fields["CheckoutEnabled"] = 239] = "CheckoutEnabled";
    fields[fields["CheckoutInstructions"] = 240] = "CheckoutInstructions";
    fields[fields["CheckoutStatus"] = 241] = "CheckoutStatus";
    fields[fields["ChildCategory"] = 242] = "ChildCategory";
    fields[fields["ChildFolder"] = 243] = "ChildFolder";
    fields[fields["CIPBankAccountStored"] = 244] = "CIPBankAccountStored";
    fields[fields["CityName"] = 245] = "CityName";
    fields[fields["ClassifiedAdAutoAcceptEnabled"] = 246] = "ClassifiedAdAutoAcceptEnabled";
    fields[fields["ClassifiedAdAutoDeclineEnabled"] = 247] = "ClassifiedAdAutoDeclineEnabled";
    fields[fields["ClassifiedAdBestOfferEnabled"] = 248] = "ClassifiedAdBestOfferEnabled";
    fields[fields["ClassifiedAdCompanyNameEnabled"] = 249] = "ClassifiedAdCompanyNameEnabled";
    fields[fields["ClassifiedAdContactByAddressEnabled"] = 250] = "ClassifiedAdContactByAddressEnabled";
    fields[fields["ClassifiedAdContactByEmailEnabled"] = 251] = "ClassifiedAdContactByEmailEnabled";
    fields[fields["ClassifiedAdContactByPhoneEnabled"] = 252] = "ClassifiedAdContactByPhoneEnabled";
    fields[fields["ClassifiedAdCount"] = 253] = "ClassifiedAdCount";
    fields[fields["ClassifiedAdCounterOfferEnabled"] = 254] = "ClassifiedAdCounterOfferEnabled";
    fields[fields["ClassifiedAdOfferCount"] = 255] = "ClassifiedAdOfferCount";
    fields[fields["ClassifiedAdPaymentMethodEnabled"] = 256] = "ClassifiedAdPaymentMethodEnabled";
    fields[fields["ClassifiedAdPayPerLeadEnabled"] = 257] = "ClassifiedAdPayPerLeadEnabled";
    fields[fields["ClassifiedAdPhoneCount"] = 258] = "ClassifiedAdPhoneCount";
    fields[fields["ClassifiedAdShippingMethodEnabled"] = 259] = "ClassifiedAdShippingMethodEnabled";
    fields[fields["ClassifiedAdStreetCount"] = 260] = "ClassifiedAdStreetCount";
    fields[fields["ClientAlertsAuthToken"] = 261] = "ClientAlertsAuthToken";
    fields[fields["CODCost"] = 262] = "CODCost";
    fields[fields["Code"] = 263] = "Code";
    fields[fields["CodiceFiscale"] = 264] = "CodiceFiscale";
    fields[fields["CODService"] = 265] = "CODService";
    fields[fields["Color"] = 266] = "Color";
    fields[fields["ColorScheme"] = 267] = "ColorScheme";
    fields[fields["ColorSchemeID"] = 268] = "ColorSchemeID";
    fields[fields["CombinedDuration"] = 269] = "CombinedDuration";
    fields[fields["CombinedFixedPriceTreatmentEnabled"] = 270] = "CombinedFixedPriceTreatmentEnabled";
    fields[fields["CombinedPaymentOption"] = 271] = "CombinedPaymentOption";
    fields[fields["CombinedPaymentPreferences"] = 272] = "CombinedPaymentPreferences";
    fields[fields["Comment"] = 273] = "Comment";
    fields[fields["CommentingUser"] = 274] = "CommentingUser";
    fields[fields["CommentingUserScore"] = 275] = "CommentingUserScore";
    fields[fields["CommentReplaced"] = 276] = "CommentReplaced";
    fields[fields["Comments"] = 277] = "Comments";
    fields[fields["CommentText"] = 278] = "CommentText";
    fields[fields["CommentTime"] = 279] = "CommentTime";
    fields[fields["CommentType"] = 280] = "CommentType";
    fields[fields["Commission"] = 281] = "Commission";
    fields[fields["CompanyName"] = 282] = "CompanyName";
    fields[fields["Compatibility"] = 283] = "Compatibility";
    fields[fields["CompatibilityNotes"] = 284] = "CompatibilityNotes";
    fields[fields["CompatibleVehicleType"] = 285] = "CompatibleVehicleType";
    fields[fields["CompleteStatus"] = 286] = "CompleteStatus";
    fields[fields["Condition"] = 287] = "Condition";
    fields[fields["ConditionDescription"] = 288] = "ConditionDescription";
    fields[fields["ConditionDisplayName"] = 289] = "ConditionDisplayName";
    fields[fields["ConditionEnabled"] = 290] = "ConditionEnabled";
    fields[fields["ConditionHelpURL"] = 291] = "ConditionHelpURL";
    fields[fields["ConditionID"] = 292] = "ConditionID";
    fields[fields["ConditionValues"] = 293] = "ConditionValues";
    fields[fields["Confidence"] = 294] = "Confidence";
    fields[fields["ContactAddress"] = 295] = "ContactAddress";
    fields[fields["ContactHoursDetails"] = 296] = "ContactHoursDetails";
    fields[fields["ContactID"] = 297] = "ContactID";
    fields[fields["ContainingOrder"] = 298] = "ContainingOrder";
    fields[fields["Content"] = 299] = "Content";
    fields[fields["ContextSearchAsset"] = 300] = "ContextSearchAsset";
    fields[fields["ConversionRate"] = 301] = "ConversionRate";
    fields[fields["ConvertedAdjustmentAmount"] = 302] = "ConvertedAdjustmentAmount";
    fields[fields["ConvertedAmountPaid"] = 303] = "ConvertedAmountPaid";
    fields[fields["ConvertedBuyItNowPrice"] = 304] = "ConvertedBuyItNowPrice";
    fields[fields["ConvertedCurrentPrice"] = 305] = "ConvertedCurrentPrice";
    fields[fields["ConvertedMaxBid"] = 306] = "ConvertedMaxBid";
    fields[fields["ConvertedPrice"] = 307] = "ConvertedPrice";
    fields[fields["ConvertedReservePrice"] = 308] = "ConvertedReservePrice";
    fields[fields["ConvertedStartPrice"] = 309] = "ConvertedStartPrice";
    fields[fields["ConvertedTransactionPrice"] = 310] = "ConvertedTransactionPrice";
    fields[fields["CopyEmail"] = 311] = "CopyEmail";
    fields[fields["CopyEmailToRightsOwner"] = 312] = "CopyEmailToRightsOwner";
    fields[fields["Copyright"] = 313] = "Copyright";
    fields[fields["CorrelationID"] = 314] = "CorrelationID";
    fields[fields["CostGroupFlat"] = 315] = "CostGroupFlat";
    fields[fields["Count"] = 316] = "Count";
    fields[fields["Countable"] = 317] = "Countable";
    fields[fields["CounterOfferPrice"] = 318] = "CounterOfferPrice";
    fields[fields["CounterOfferQuantity"] = 319] = "CounterOfferQuantity";
    fields[fields["Country"] = 320] = "Country";
    fields[fields["CountryDetails"] = 321] = "CountryDetails";
    fields[fields["CountryName"] = 322] = "CountryName";
    fields[fields["CountryOfOrigin"] = 323] = "CountryOfOrigin";
    fields[fields["CountsTowardAggregate"] = 324] = "CountsTowardAggregate";
    fields[fields["County"] = 325] = "County";
    fields[fields["CreatedDate"] = 326] = "CreatedDate";
    fields[fields["CreatedTime"] = 327] = "CreatedTime";
    fields[fields["CreateTimeFrom"] = 328] = "CreateTimeFrom";
    fields[fields["CreateTimeTo"] = 329] = "CreateTimeTo";
    fields[fields["CreatingUserRole"] = 330] = "CreatingUserRole";
    fields[fields["CreationDate"] = 331] = "CreationDate";
    fields[fields["CreationTime"] = 332] = "CreationTime";
    fields[fields["CreditCardExpiration"] = 333] = "CreditCardExpiration";
    fields[fields["CreditCardInfo"] = 334] = "CreditCardInfo";
    fields[fields["CreditCardModifyDate"] = 335] = "CreditCardModifyDate";
    fields[fields["CrossBorderTrade"] = 336] = "CrossBorderTrade";
    fields[fields["CrossBorderTradeAustraliaEnabled"] = 337] = "CrossBorderTradeAustraliaEnabled";
    fields[fields["CrossBorderTradeGBEnabled"] = 338] = "CrossBorderTradeGBEnabled";
    fields[fields["CrossBorderTradeNorthAmericaEnabled"] = 339] = "CrossBorderTradeNorthAmericaEnabled";
    fields[fields["CrossBorderTransactionCount"] = 340] = "CrossBorderTransactionCount";
    fields[fields["CrossBorderTransactionPercent"] = 341] = "CrossBorderTransactionPercent";
    fields[fields["CrossPromotion"] = 342] = "CrossPromotion";
    fields[fields["CrossPromotionEnabled"] = 343] = "CrossPromotionEnabled";
    fields[fields["CrossPromotionPreferences"] = 344] = "CrossPromotionPreferences";
    fields[fields["CrossSellGallerySortFilter"] = 345] = "CrossSellGallerySortFilter";
    fields[fields["CrossSellItemFormatSortFilter"] = 346] = "CrossSellItemFormatSortFilter";
    fields[fields["CrossSellItemSortFilter"] = 347] = "CrossSellItemSortFilter";
    fields[fields["Currency"] = 348] = "Currency";
    fields[fields["CurrencyDetails"] = 349] = "CurrencyDetails";
    fields[fields["CurrencyID"] = 350] = "CurrencyID";
    fields[fields["CurrentBalance"] = 351] = "CurrentBalance";
    fields[fields["CurrentFieldText"] = 352] = "CurrentFieldText";
    fields[fields["CurrentFieldValue"] = 353] = "CurrentFieldValue";
    fields[fields["CurrentPrice"] = 354] = "CurrentPrice";
    fields[fields["CustomCategories"] = 355] = "CustomCategories";
    fields[fields["CustomCategory"] = 356] = "CustomCategory";
    fields[fields["CustomEmailName"] = 357] = "CustomEmailName";
    fields[fields["CustomHeader"] = 358] = "CustomHeader";
    fields[fields["CustomHeaderLayout"] = 359] = "CustomHeaderLayout";
    fields[fields["CustomLabel"] = 360] = "CustomLabel";
    fields[fields["CustomListingHeader"] = 361] = "CustomListingHeader";
    fields[fields["CustomPage"] = 362] = "CustomPage";
    fields[fields["CustomPageArray"] = 363] = "CustomPageArray";
    fields[fields["CutoffTime"] = 364] = "CutoffTime";
    fields[fields["DailyHardLimit"] = 365] = "DailyHardLimit";
    fields[fields["DailySoftLimit"] = 366] = "DailySoftLimit";
    fields[fields["DailyUsage"] = 367] = "DailyUsage";
    fields[fields["Date"] = 368] = "Date";
    fields[fields["DaylightSavingsInEffect"] = 369] = "DaylightSavingsInEffect";
    fields[fields["DaylightSavingsLabel"] = 370] = "DaylightSavingsLabel";
    fields[fields["DaylightSavingsOffset"] = 371] = "DaylightSavingsOffset";
    fields[fields["DayOfWeek"] = 372] = "DayOfWeek";
    fields[fields["DaysSinceSale"] = 373] = "DaysSinceSale";
    fields[fields["DaysToFullPayment"] = 374] = "DaysToFullPayment";
    fields[fields["DeclinedRTERequestCount"] = 375] = "DeclinedRTERequestCount";
    fields[fields["DefaultPayPalEmailAddress"] = 376] = "DefaultPayPalEmailAddress";
    fields[fields["DefaultValue"] = 377] = "DefaultValue";
    fields[fields["DelayBeforeOpeningDispute"] = 378] = "DelayBeforeOpeningDispute";
    fields[fields["Delete"] = 379] = "Delete";
    fields[fields["DeleteAutomatedListingRule"] = 380] = "DeleteAutomatedListingRule";
    fields[fields["DeleteAutomatedRelistingRule"] = 381] = "DeleteAutomatedRelistingRule";
    fields[fields["DeleteAutomatedSecondChanceOfferRule"] = 382] = "DeleteAutomatedSecondChanceOfferRule";
    fields[fields["DeletedField"] = 383] = "DeletedField";
    fields[fields["DeletedFromLostList"] = 384] = "DeletedFromLostList";
    fields[fields["DeletedFromSoldList"] = 385] = "DeletedFromSoldList";
    fields[fields["DeletedFromUnsoldList"] = 386] = "DeletedFromUnsoldList";
    fields[fields["DeletedFromWonList"] = 387] = "DeletedFromWonList";
    fields[fields["DeletedSaleTemplateID"] = 388] = "DeletedSaleTemplateID";
    fields[fields["DeletedSaleTemplateName"] = 389] = "DeletedSaleTemplateName";
    fields[fields["DeletedSellingManagerProductDetails"] = 390] = "DeletedSellingManagerProductDetails";
    fields[fields["DeliveredCount"] = 391] = "DeliveredCount";
    fields[fields["DeliveryStatus"] = 392] = "DeliveryStatus";
    fields[fields["DeliveryTime"] = 393] = "DeliveryTime";
    fields[fields["DeliveryURL"] = 394] = "DeliveryURL";
    fields[fields["DeliveryURLDetails"] = 395] = "DeliveryURLDetails";
    fields[fields["DeliveryURLName"] = 396] = "DeliveryURLName";
    fields[fields["DepositAmount"] = 397] = "DepositAmount";
    fields[fields["DepositSupported"] = 398] = "DepositSupported";
    fields[fields["DepositType"] = 399] = "DepositType";
    fields[fields["DeprecationDetails"] = 400] = "DeprecationDetails";
    fields[fields["DescColor"] = 401] = "DescColor";
    fields[fields["DescFace"] = 402] = "DescFace";
    fields[fields["Description"] = 403] = "Description";
    fields[fields["DescriptionReviseMode"] = 404] = "DescriptionReviseMode";
    fields[fields["DescriptionTemplate"] = 405] = "DescriptionTemplate";
    fields[fields["DescSize"] = 406] = "DescSize";
    fields[fields["DestinationCountryCode"] = 407] = "DestinationCountryCode";
    fields[fields["DestinationParentCategoryID"] = 408] = "DestinationParentCategoryID";
    fields[fields["DestinationPostalCode"] = 409] = "DestinationPostalCode";
    fields[fields["DetailedMessage"] = 410] = "DetailedMessage";
    fields[fields["DetailedText"] = 411] = "DetailedText";
    fields[fields["DetailLevel"] = 412] = "DetailLevel";
    fields[fields["DetailName"] = 413] = "DetailName";
    fields[fields["DetailsURL"] = 414] = "DetailsURL";
    fields[fields["DetailVersion"] = 415] = "DetailVersion";
    fields[fields["DeviceType"] = 416] = "DeviceType";
    fields[fields["DimensionsRequired"] = 417] = "DimensionsRequired";
    fields[fields["DimensionsSupported"] = 418] = "DimensionsSupported";
    fields[fields["DisableBuyerRequirements"] = 419] = "DisableBuyerRequirements";
    fields[fields["DiscountName"] = 420] = "DiscountName";
    fields[fields["DiscountPriceInfo"] = 421] = "DiscountPriceInfo";
    fields[fields["DiscountProfile"] = 422] = "DiscountProfile";
    fields[fields["DiscountProfileID"] = 423] = "DiscountProfileID";
    fields[fields["DiscountProfileName"] = 424] = "DiscountProfileName";
    fields[fields["DiscountReason"] = 425] = "DiscountReason";
    fields[fields["DiscountType"] = 426] = "DiscountType";
    fields[fields["DiscountValue"] = 427] = "DiscountValue";
    fields[fields["DispatchCutoffTimePreference"] = 428] = "DispatchCutoffTimePreference";
    fields[fields["DispatchTimeMax"] = 429] = "DispatchTimeMax";
    fields[fields["DispatchTimeMaxDetails"] = 430] = "DispatchTimeMaxDetails";
    fields[fields["DisplayLogoSelling"] = 431] = "DisplayLogoSelling";
    fields[fields["DisplayMessageStoreCustomText"] = 432] = "DisplayMessageStoreCustomText";
    fields[fields["DisplayName"] = 433] = "DisplayName";
    fields[fields["DisplayPayNowButton"] = 434] = "DisplayPayNowButton";
    fields[fields["DisplayToPublic"] = 435] = "DisplayToPublic";
    fields[fields["DisplayType"] = 436] = "DisplayType";
    fields[fields["Dispute"] = 437] = "Dispute";
    fields[fields["DisputeActivity"] = 438] = "DisputeActivity";
    fields[fields["DisputeArray"] = 439] = "DisputeArray";
    fields[fields["DisputeCreatedTime"] = 440] = "DisputeCreatedTime";
    fields[fields["DisputeCreditEligibility"] = 441] = "DisputeCreditEligibility";
    fields[fields["DisputeExplanation"] = 442] = "DisputeExplanation";
    fields[fields["DisputeFilterCount"] = 443] = "DisputeFilterCount";
    fields[fields["DisputeFilterType"] = 444] = "DisputeFilterType";
    fields[fields["DisputeID"] = 445] = "DisputeID";
    fields[fields["DisputeMessage"] = 446] = "DisputeMessage";
    fields[fields["DisputeModifiedTime"] = 447] = "DisputeModifiedTime";
    fields[fields["DisputeReason"] = 448] = "DisputeReason";
    fields[fields["DisputeRecordType"] = 449] = "DisputeRecordType";
    fields[fields["DisputeResolution"] = 450] = "DisputeResolution";
    fields[fields["DisputeResolutionReason"] = 451] = "DisputeResolutionReason";
    fields[fields["DisputeResolutionRecordType"] = 452] = "DisputeResolutionRecordType";
    fields[fields["DisputeSortType"] = 453] = "DisputeSortType";
    fields[fields["DisputeState"] = 454] = "DisputeState";
    fields[fields["DisputeStatus"] = 455] = "DisputeStatus";
    fields[fields["DocsForCCProcessingToSendCount"] = 456] = "DocsForCCProcessingToSendCount";
    fields[fields["DomesticRateTable"] = 457] = "DomesticRateTable";
    fields[fields["DonationPercent"] = 458] = "DonationPercent";
    fields[fields["DuplicateInvocationDetails"] = 459] = "DuplicateInvocationDetails";
    fields[fields["DuplicateInvocationID"] = 460] = "DuplicateInvocationID";
    fields[fields["Duration"] = 461] = "Duration";
    fields[fields["DurationInDays"] = 462] = "DurationInDays";
    fields[fields["DutchBINEnabled"] = 463] = "DutchBINEnabled";
    fields[fields["EachAdditionalAmount"] = 464] = "EachAdditionalAmount";
    fields[fields["EachAdditionalAmountOff"] = 465] = "EachAdditionalAmountOff";
    fields[fields["EachAdditionalOffAmount"] = 466] = "EachAdditionalOffAmount";
    fields[fields["EachAdditionalPercentOff"] = 467] = "EachAdditionalPercentOff";
    fields[fields["EAN"] = 468] = "EAN";
    fields[fields["EANIdentifierEnabled"] = 469] = "EANIdentifierEnabled";
    fields[fields["eBayAuthToken"] = 470] = "eBayAuthToken";
    fields[fields["eBayGoodStanding"] = 471] = "eBayGoodStanding";
    fields[fields["eBayMotorsProAdFormatEnabled"] = 472] = "eBayMotorsProAdFormatEnabled";
    fields[fields["eBayMotorsProAutoAcceptEnabled"] = 473] = "eBayMotorsProAutoAcceptEnabled";
    fields[fields["eBayMotorsProAutoDeclineEnabled"] = 474] = "eBayMotorsProAutoDeclineEnabled";
    fields[fields["eBayMotorsProBestOfferEnabled"] = 475] = "eBayMotorsProBestOfferEnabled";
    fields[fields["eBayMotorsProCompanyNameEnabled"] = 476] = "eBayMotorsProCompanyNameEnabled";
    fields[fields["eBayMotorsProContactByAddressEnabled"] = 477] = "eBayMotorsProContactByAddressEnabled";
    fields[fields["eBayMotorsProContactByEmailEnabled"] = 478] = "eBayMotorsProContactByEmailEnabled";
    fields[fields["eBayMotorsProContactByPhoneEnabled"] = 479] = "eBayMotorsProContactByPhoneEnabled";
    fields[fields["eBayMotorsProCounterOfferEnabled"] = 480] = "eBayMotorsProCounterOfferEnabled";
    fields[fields["eBayMotorsProPaymentMethodCheckOutEnabled"] = 481] = "eBayMotorsProPaymentMethodCheckOutEnabled";
    fields[fields["eBayMotorsProPhoneCount"] = 482] = "eBayMotorsProPhoneCount";
    fields[fields["eBayMotorsProSellerContactDetailsEnabled"] = 483] = "eBayMotorsProSellerContactDetailsEnabled";
    fields[fields["eBayMotorsProShippingMethodEnabled"] = 484] = "eBayMotorsProShippingMethodEnabled";
    fields[fields["eBayMotorsProStreetCount"] = 485] = "eBayMotorsProStreetCount";
    fields[fields["eBayNotes"] = 486] = "eBayNotes";
    fields[fields["eBayNowEligible"] = 487] = "eBayNowEligible";
    fields[fields["eBayPaymentMismatchDetails"] = 488] = "eBayPaymentMismatchDetails";
    fields[fields["eBayPaymentStatus"] = 489] = "eBayPaymentStatus";
    fields[fields["eBayWikiReadOnly"] = 490] = "eBayWikiReadOnly";
    fields[fields["EIASToken"] = 491] = "EIASToken";
    fields[fields["EligibleForPickupInStore"] = 492] = "EligibleForPickupInStore";
    fields[fields["Email"] = 493] = "Email";
    fields[fields["EmailCopyToSeller"] = 494] = "EmailCopyToSeller";
    fields[fields["EmailCopyToSender"] = 495] = "EmailCopyToSender";
    fields[fields["EmailCustomized"] = 496] = "EmailCustomized";
    fields[fields["EmailDateRange"] = 497] = "EmailDateRange";
    fields[fields["EmailLog"] = 498] = "EmailLog";
    fields[fields["EmailShipmentTrackingNumberPreference"] = 499] = "EmailShipmentTrackingNumberPreference";
    fields[fields["EmailsSent"] = 500] = "EmailsSent";
    fields[fields["EmailState"] = 501] = "EmailState";
    fields[fields["EmailType"] = 502] = "EmailType";
    fields[fields["Encoding"] = 503] = "Encoding";
    fields[fields["EncryptedID"] = 504] = "EncryptedID";
    fields[fields["EndCreationTime"] = 505] = "EndCreationTime";
    fields[fields["EndDate"] = 506] = "EndDate";
    fields[fields["EndingDisputeID"] = 507] = "EndingDisputeID";
    fields[fields["EndingReason"] = 508] = "EndingReason";
    fields[fields["EndItemRequestContainer"] = 509] = "EndItemRequestContainer";
    fields[fields["EndItemResponseContainer"] = 510] = "EndItemResponseContainer";
    fields[fields["EndOfAuctionEmailPreferences"] = 511] = "EndOfAuctionEmailPreferences";
    fields[fields["EndTime"] = 512] = "EndTime";
    fields[fields["EndTimeFrom"] = 513] = "EndTimeFrom";
    fields[fields["EndTimeTo"] = 514] = "EndTimeTo";
    fields[fields["EndUserIP"] = 515] = "EndUserIP";
    fields[fields["EnforcedOnListingSite"] = 516] = "EnforcedOnListingSite";
    fields[fields["EnforcedOnRegistrationSite"] = 517] = "EnforcedOnRegistrationSite";
    fields[fields["EnterpriseSeller"] = 518] = "EnterpriseSeller";
    fields[fields["EntriesPerPage"] = 519] = "EntriesPerPage";
    fields[fields["EPID"] = 520] = "EPID";
    fields[fields["ErrorClassification"] = 521] = "ErrorClassification";
    fields[fields["ErrorCode"] = 522] = "ErrorCode";
    fields[fields["ErrorCount"] = 523] = "ErrorCount";
    fields[fields["ErrorHandling"] = 524] = "ErrorHandling";
    fields[fields["ErrorLanguage"] = 525] = "ErrorLanguage";
    fields[fields["ErrorMessage"] = 526] = "ErrorMessage";
    fields[fields["ErrorParameters"] = 527] = "ErrorParameters";
    fields[fields["Errors"] = 528] = "Errors";
    fields[fields["Escalation"] = 529] = "Escalation";
    fields[fields["EstimatedRefundCompletionTime"] = 530] = "EstimatedRefundCompletionTime";
    fields[fields["EventEnable"] = 531] = "EventEnable";
    fields[fields["EventProperty"] = 532] = "EventProperty";
    fields[fields["EventTime"] = 533] = "EventTime";
    fields[fields["EventTitle"] = 534] = "EventTitle";
    fields[fields["EventType"] = 535] = "EventType";
    fields[fields["ExcludeBalance"] = 536] = "ExcludeBalance";
    fields[fields["ExcludeCategoryID"] = 537] = "ExcludeCategoryID";
    fields[fields["ExcludedUser"] = 538] = "ExcludedUser";
    fields[fields["ExcludeRelationships"] = 539] = "ExcludeRelationships";
    fields[fields["ExcludeShippingLocationDetails"] = 540] = "ExcludeShippingLocationDetails";
    fields[fields["ExcludeShipToLocation"] = 541] = "ExcludeShipToLocation";
    fields[fields["ExcludeSummary"] = 542] = "ExcludeSummary";
    fields[fields["ExpectedReleaseDate"] = 543] = "ExpectedReleaseDate";
    fields[fields["ExpeditedService"] = 544] = "ExpeditedService";
    fields[fields["ExpirationDate"] = 545] = "ExpirationDate";
    fields[fields["ExpirationTime"] = 546] = "ExpirationTime";
    fields[fields["Expired"] = 547] = "Expired";
    fields[fields["ExpiredCount"] = 548] = "ExpiredCount";
    fields[fields["ExportListings"] = 549] = "ExportListings";
    fields[fields["ExpressConditionRequired"] = 550] = "ExpressConditionRequired";
    fields[fields["ExpressEnabled"] = 551] = "ExpressEnabled";
    fields[fields["ExpressPicturesRequired"] = 552] = "ExpressPicturesRequired";
    fields[fields["ExtendedHandling"] = 553] = "ExtendedHandling";
    fields[fields["ExtendedSellerContactDetails"] = 554] = "ExtendedSellerContactDetails";
    fields[fields["ExtensionInDays"] = 555] = "ExtensionInDays";
    fields[fields["ExternalAddressID"] = 556] = "ExternalAddressID";
    fields[fields["ExternalEmail"] = 557] = "ExternalEmail";
    fields[fields["ExternalID"] = 558] = "ExternalID";
    fields[fields["ExternalMessageID"] = 559] = "ExternalMessageID";
    fields[fields["ExternalMessageIDs"] = 560] = "ExternalMessageIDs";
    fields[fields["ExternalPictureURL"] = 561] = "ExternalPictureURL";
    fields[fields["ExternalProductID"] = 562] = "ExternalProductID";
    fields[fields["ExternalReferenceID"] = 563] = "ExternalReferenceID";
    fields[fields["ExternalTransaction"] = 564] = "ExternalTransaction";
    fields[fields["ExternalTransactionID"] = 565] = "ExternalTransactionID";
    fields[fields["ExternalTransactionStatus"] = 566] = "ExternalTransactionStatus";
    fields[fields["ExternalTransactionTime"] = 567] = "ExternalTransactionTime";
    fields[fields["ExternalUserData"] = 568] = "ExternalUserData";
    fields[fields["FavoriteItemID"] = 569] = "FavoriteItemID";
    fields[fields["FavoriteSearch"] = 570] = "FavoriteSearch";
    fields[fields["FavoriteSearches"] = 571] = "FavoriteSearches";
    fields[fields["FavoriteSeller"] = 572] = "FavoriteSeller";
    fields[fields["FavoriteSellerCount"] = 573] = "FavoriteSellerCount";
    fields[fields["FavoriteSellers"] = 574] = "FavoriteSellers";
    fields[fields["Fax"] = 575] = "Fax";
    fields[fields["Featured"] = 576] = "Featured";
    fields[fields["FeatureDefinitions"] = 577] = "FeatureDefinitions";
    fields[fields["FeaturedFirst"] = 578] = "FeaturedFirst";
    fields[fields["FeaturedPlus"] = 579] = "FeaturedPlus";
    fields[fields["FeatureEligibility"] = 580] = "FeatureEligibility";
    fields[fields["FeatureID"] = 581] = "FeatureID";
    fields[fields["FedExRateOption"] = 582] = "FedExRateOption";
    fields[fields["Fee"] = 583] = "Fee";
    fields[fields["FeedbackDetail"] = 584] = "FeedbackDetail";
    fields[fields["FeedbackDetailArray"] = 585] = "FeedbackDetailArray";
    fields[fields["FeedbackDetailItemTotal"] = 586] = "FeedbackDetailItemTotal";
    fields[fields["FeedbackID"] = 587] = "FeedbackID";
    fields[fields["FeedbackInfo"] = 588] = "FeedbackInfo";
    fields[fields["FeedbackLeft"] = 589] = "FeedbackLeft";
    fields[fields["FeedbackLeftPercent"] = 590] = "FeedbackLeftPercent";
    fields[fields["FeedbackPeriod"] = 591] = "FeedbackPeriod";
    fields[fields["FeedbackPrivate"] = 592] = "FeedbackPrivate";
    fields[fields["FeedbackRatingStar"] = 593] = "FeedbackRatingStar";
    fields[fields["FeedbackReceived"] = 594] = "FeedbackReceived";
    fields[fields["FeedbackResponse"] = 595] = "FeedbackResponse";
    fields[fields["FeedbackRevised"] = 596] = "FeedbackRevised";
    fields[fields["FeedbackScore"] = 597] = "FeedbackScore";
    fields[fields["FeedbackSent"] = 598] = "FeedbackSent";
    fields[fields["FeedbackSummary"] = 599] = "FeedbackSummary";
    fields[fields["FeedbackSummaryPeriod"] = 600] = "FeedbackSummaryPeriod";
    fields[fields["FeedbackToReceiveCount"] = 601] = "FeedbackToReceiveCount";
    fields[fields["FeedbackToSendCount"] = 602] = "FeedbackToSendCount";
    fields[fields["FeedbackType"] = 603] = "FeedbackType";
    fields[fields["FeeOrCreditAmount"] = 604] = "FeeOrCreditAmount";
    fields[fields["Fees"] = 605] = "Fees";
    fields[fields["Field"] = 606] = "Field";
    fields[fields["FieldName"] = 607] = "FieldName";
    fields[fields["FieldTip"] = 608] = "FieldTip";
    fields[fields["FileReferenceID"] = 609] = "FileReferenceID";
    fields[fields["Filter"] = 610] = "Filter";
    fields[fields["FinalValueFee"] = 611] = "FinalValueFee";
    fields[fields["FinancingAnswer"] = 612] = "FinancingAnswer";
    fields[fields["FirstName"] = 613] = "FirstName";
    fields[fields["Flagged"] = 614] = "Flagged";
    fields[fields["FlaggedMessageCount"] = 615] = "FlaggedMessageCount";
    fields[fields["FlatRateInsuranceRange"] = 616] = "FlatRateInsuranceRange";
    fields[fields["FlatRateInsuranceRangeCost"] = 617] = "FlatRateInsuranceRangeCost";
    fields[fields["FlatShippingDiscount"] = 618] = "FlatShippingDiscount";
    fields[fields["FloorPrice"] = 619] = "FloorPrice";
    fields[fields["Folder"] = 620] = "Folder";
    fields[fields["FolderComment"] = 621] = "FolderComment";
    fields[fields["FolderID"] = 622] = "FolderID";
    fields[fields["FolderLevel"] = 623] = "FolderLevel";
    fields[fields["FolderName"] = 624] = "FolderName";
    fields[fields["FolderSummary"] = 625] = "FolderSummary";
    fields[fields["Followup"] = 626] = "Followup";
    fields[fields["FollowUpReplaced"] = 627] = "FollowUpReplaced";
    fields[fields["Font"] = 628] = "Font";
    fields[fields["ForwardDetails"] = 629] = "ForwardDetails";
    fields[fields["FreeAddedCategory"] = 630] = "FreeAddedCategory";
    fields[fields["FreeGalleryPlusEnabled"] = 631] = "FreeGalleryPlusEnabled";
    fields[fields["FreePicturePackEnabled"] = 632] = "FreePicturePackEnabled";
    fields[fields["FreeShipping"] = 633] = "FreeShipping";
    fields[fields["Frequency"] = 634] = "Frequency";
    fields[fields["FullRecursion"] = 635] = "FullRecursion";
    fields[fields["FullURL"] = 636] = "FullURL";
    fields[fields["GalleryDuration"] = 637] = "GalleryDuration";
    fields[fields["GalleryErrorInfo"] = 638] = "GalleryErrorInfo";
    fields[fields["GalleryFeaturedDurations"] = 639] = "GalleryFeaturedDurations";
    fields[fields["GalleryStatus"] = 640] = "GalleryStatus";
    fields[fields["GalleryType"] = 641] = "GalleryType";
    fields[fields["GalleryURL"] = 642] = "GalleryURL";
    fields[fields["GeneralAlert"] = 643] = "GeneralAlert";
    fields[fields["GenericColorSchemeArray"] = 644] = "GenericColorSchemeArray";
    fields[fields["GetItFast"] = 645] = "GetItFast";
    fields[fields["GetRecommendationsRequestContainer"] = 646] = "GetRecommendationsRequestContainer";
    fields[fields["GetRecommendationsResponseContainer"] = 647] = "GetRecommendationsResponseContainer";
    fields[fields["GiftIcon"] = 648] = "GiftIcon";
    fields[fields["GiftServices"] = 649] = "GiftServices";
    fields[fields["GlobalShipping"] = 650] = "GlobalShipping";
    fields[fields["GlobalShippingEnabled"] = 651] = "GlobalShippingEnabled";
    fields[fields["GlobalShippingProgramListingPreference"] = 652] = "GlobalShippingProgramListingPreference";
    fields[fields["GoodStanding"] = 653] = "GoodStanding";
    fields[fields["GranularityLevel"] = 654] = "GranularityLevel";
    fields[fields["GrossDetailAmount"] = 655] = "GrossDetailAmount";
    fields[fields["Group"] = 656] = "Group";
    fields[fields["Group1MaxFlatShippingCost"] = 657] = "Group1MaxFlatShippingCost";
    fields[fields["Group2MaxFlatShippingCost"] = 658] = "Group2MaxFlatShippingCost";
    fields[fields["Group3MaxFlatShippingCost"] = 659] = "Group3MaxFlatShippingCost";
    fields[fields["GroupID"] = 660] = "GroupID";
    fields[fields["GroupName"] = 661] = "GroupName";
    fields[fields["GTIN"] = 662] = "GTIN";
    fields[fields["HandlingTimeEnabled"] = 663] = "HandlingTimeEnabled";
    fields[fields["HardExpirationTime"] = 664] = "HardExpirationTime";
    fields[fields["HardExpirationWarning"] = 665] = "HardExpirationWarning";
    fields[fields["HasMoreEntries"] = 666] = "HasMoreEntries";
    fields[fields["HasMoreItems"] = 667] = "HasMoreItems";
    fields[fields["HasMoreOrders"] = 668] = "HasMoreOrders";
    fields[fields["HasMorePayments"] = 669] = "HasMorePayments";
    fields[fields["HasMoreTransactions"] = 670] = "HasMoreTransactions";
    fields[fields["HasPublicMessages"] = 671] = "HasPublicMessages";
    fields[fields["HasReservePrice"] = 672] = "HasReservePrice";
    fields[fields["HasUnansweredQuestions"] = 673] = "HasUnansweredQuestions";
    fields[fields["HeaderStyle"] = 674] = "HeaderStyle";
    fields[fields["HelpText"] = 675] = "HelpText";
    fields[fields["HelpURL"] = 676] = "HelpURL";
    fields[fields["HelpURLPath"] = 677] = "HelpURLPath";
    fields[fields["HideFixedPriceStoreItems"] = 678] = "HideFixedPriceStoreItems";
    fields[fields["HideFromSearch"] = 679] = "HideFromSearch";
    fields[fields["HideSendersEmailAddress"] = 680] = "HideSendersEmailAddress";
    fields[fields["HideVariations"] = 681] = "HideVariations";
    fields[fields["HighBidder"] = 682] = "HighBidder";
    fields[fields["HighestBid"] = 683] = "HighestBid";
    fields[fields["Highlight"] = 684] = "Highlight";
    fields[fields["HighPriority"] = 685] = "HighPriority";
    fields[fields["HitCount"] = 686] = "HitCount";
    fields[fields["HitCounter"] = 687] = "HitCounter";
    fields[fields["HomePage"] = 688] = "HomePage";
    fields[fields["HomePageFeatured"] = 689] = "HomePageFeatured";
    fields[fields["HomePageFeaturedEnabled"] = 690] = "HomePageFeaturedEnabled";
    fields[fields["HourlyHardLimit"] = 691] = "HourlyHardLimit";
    fields[fields["HourlySoftLimit"] = 692] = "HourlySoftLimit";
    fields[fields["HourlyUsage"] = 693] = "HourlyUsage";
    fields[fields["Hours1AnyTime"] = 694] = "Hours1AnyTime";
    fields[fields["Hours1Days"] = 695] = "Hours1Days";
    fields[fields["Hours1From"] = 696] = "Hours1From";
    fields[fields["Hours1To"] = 697] = "Hours1To";
    fields[fields["Hours2AnyTime"] = 698] = "Hours2AnyTime";
    fields[fields["Hours2Days"] = 699] = "Hours2Days";
    fields[fields["Hours2From"] = 700] = "Hours2From";
    fields[fields["Hours2To"] = 701] = "Hours2To";
    fields[fields["HoursToDeposit"] = 702] = "HoursToDeposit";
    fields[fields["ID"] = 703] = "ID";
    fields[fields["IDVerified"] = 704] = "IDVerified";
    fields[fields["ImageChallengeURL"] = 705] = "ImageChallengeURL";
    fields[fields["ImageURL"] = 706] = "ImageURL";
    fields[fields["ImmediatePayEligible"] = 707] = "ImmediatePayEligible";
    fields[fields["ImportCharge"] = 708] = "ImportCharge";
    fields[fields["Imposition"] = 709] = "Imposition";
    fields[fields["Include"] = 710] = "Include";
    fields[fields["IncludeASQPreferences"] = 711] = "IncludeASQPreferences";
    fields[fields["IncludeBiddingSummary"] = 712] = "IncludeBiddingSummary";
    fields[fields["IncludeCodiceFiscale"] = 713] = "IncludeCodiceFiscale";
    fields[fields["IncludeConfidence"] = 714] = "IncludeConfidence";
    fields[fields["IncludeContainingOrder"] = 715] = "IncludeContainingOrder";
    fields[fields["IncludeConversionRate"] = 716] = "IncludeConversionRate";
    fields[fields["IncludeCrossPromotion"] = 717] = "IncludeCrossPromotion";
    fields[fields["IncludeDescription"] = 718] = "IncludeDescription";
    fields[fields["IncludeFavoriteSellerCount"] = 719] = "IncludeFavoriteSellerCount";
    fields[fields["IncludeFeatureEligibility"] = 720] = "IncludeFeatureEligibility";
    fields[fields["IncludeFinalValueFee"] = 721] = "IncludeFinalValueFee";
    fields[fields["IncludeFinalValueFees"] = 722] = "IncludeFinalValueFees";
    fields[fields["IncludeHighPriorityMessageOnly"] = 723] = "IncludeHighPriorityMessageOnly";
    fields[fields["IncludeItemCompatibilityList"] = 724] = "IncludeItemCompatibilityList";
    fields[fields["IncludeItemCount"] = 725] = "IncludeItemCount";
    fields[fields["IncludeItemSpecifics"] = 726] = "IncludeItemSpecifics";
    fields[fields["IncludeListContents"] = 727] = "IncludeListContents";
    fields[fields["IncludeMemberMessages"] = 728] = "IncludeMemberMessages";
    fields[fields["IncludeNotes"] = 729] = "IncludeNotes";
    fields[fields["IncludePrefilledItemInformation"] = 730] = "IncludePrefilledItemInformation";
    fields[fields["IncludeRecommendations"] = 731] = "IncludeRecommendations";
    fields[fields["IncludeReportedItemDetails"] = 732] = "IncludeReportedItemDetails";
    fields[fields["IncludeStockPhotoURL"] = 733] = "IncludeStockPhotoURL";
    fields[fields["IncludeTaxTable"] = 734] = "IncludeTaxTable";
    fields[fields["IncludeVariations"] = 735] = "IncludeVariations";
    fields[fields["IncludeVariationSpecifics"] = 736] = "IncludeVariationSpecifics";
    fields[fields["IncludeWatchCount"] = 737] = "IncludeWatchCount";
    fields[fields["INEscrowWorkflowTimeline"] = 738] = "INEscrowWorkflowTimeline";
    fields[fields["InsuranceCost"] = 739] = "InsuranceCost";
    fields[fields["InsuranceDetails"] = 740] = "InsuranceDetails";
    fields[fields["InsuranceFee"] = 741] = "InsuranceFee";
    fields[fields["InsuranceOption"] = 742] = "InsuranceOption";
    fields[fields["InsuranceType"] = 743] = "InsuranceType";
    fields[fields["InsuranceWanted"] = 744] = "InsuranceWanted";
    fields[fields["IntangibleItem"] = 745] = "IntangibleItem";
    fields[fields["IntegratedMerchantCreditCardEnabled"] = 746] = "IntegratedMerchantCreditCardEnabled";
    fields[fields["IntegratedMerchantCreditCardInfo"] = 747] = "IntegratedMerchantCreditCardInfo";
    fields[fields["InternationalCalculatedShippingDiscount"] = 748] = "InternationalCalculatedShippingDiscount";
    fields[fields["InternationalFlatShippingDiscount"] = 749] = "InternationalFlatShippingDiscount";
    fields[fields["InternationalInsuranceDetails"] = 750] = "InternationalInsuranceDetails";
    fields[fields["InternationalName"] = 751] = "InternationalName";
    fields[fields["InternationalPackagingHandlingCosts"] = 752] = "InternationalPackagingHandlingCosts";
    fields[fields["InternationalPromotionalShippingDiscount"] = 753] = "InternationalPromotionalShippingDiscount";
    fields[fields["InternationalRateTable"] = 754] = "InternationalRateTable";
    fields[fields["InternationalService"] = 755] = "InternationalService";
    fields[fields["InternationalShippingDiscountProfileID"] = 756] = "InternationalShippingDiscountProfileID";
    fields[fields["InternationalShippingInsurance"] = 757] = "InternationalShippingInsurance";
    fields[fields["InternationalShippingServiceOption"] = 758] = "InternationalShippingServiceOption";
    fields[fields["InternationalShippingServiceOptions"] = 759] = "InternationalShippingServiceOptions";
    fields[fields["InternationalStateAndCity"] = 760] = "InternationalStateAndCity";
    fields[fields["InternationalStreet"] = 761] = "InternationalStreet";
    fields[fields["IntlAutosFixedCat"] = 762] = "IntlAutosFixedCat";
    fields[fields["InventoryAlert"] = 763] = "InventoryAlert";
    fields[fields["InventoryCountLastCalculatedDate"] = 764] = "InventoryCountLastCalculatedDate";
    fields[fields["InventoryStatus"] = 765] = "InventoryStatus";
    fields[fields["InventoryTrackingMethod"] = 766] = "InventoryTrackingMethod";
    fields[fields["InvocationID"] = 767] = "InvocationID";
    fields[fields["InvocationTrackingID"] = 768] = "InvocationTrackingID";
    fields[fields["InvoiceBalance"] = 769] = "InvoiceBalance";
    fields[fields["InvoiceCredit"] = 770] = "InvoiceCredit";
    fields[fields["InvoiceDate"] = 771] = "InvoiceDate";
    fields[fields["InvoiceNewFee"] = 772] = "InvoiceNewFee";
    fields[fields["InvoiceNumber"] = 773] = "InvoiceNumber";
    fields[fields["InvoicePayment"] = 774] = "InvoicePayment";
    fields[fields["InvoiceSentTime"] = 775] = "InvoiceSentTime";
    fields[fields["ISBN"] = 776] = "ISBN";
    fields[fields["ISBNIdentifierEnabled"] = 777] = "ISBNIdentifierEnabled";
    fields[fields["IsDefault"] = 778] = "IsDefault";
    fields[fields["IsIntermediatedShippingEligible"] = 779] = "IsIntermediatedShippingEligible";
    fields[fields["IsMultiLegShipping"] = 780] = "IsMultiLegShipping";
    fields[fields["Item"] = 781] = "Item";
    fields[fields["ItemArray"] = 782] = "ItemArray";
    fields[fields["ItemBestOffers"] = 783] = "ItemBestOffers";
    fields[fields["ItemBestOffersArray"] = 784] = "ItemBestOffersArray";
    fields[fields["ItemBidDetails"] = 785] = "ItemBidDetails";
    fields[fields["ItemCompatibilityCount"] = 786] = "ItemCompatibilityCount";
    fields[fields["ItemCompatibilityEnabled"] = 787] = "ItemCompatibilityEnabled";
    fields[fields["ItemCompatibilityList"] = 788] = "ItemCompatibilityList";
    fields[fields["ItemCost"] = 789] = "ItemCost";
    fields[fields["ItemCount"] = 790] = "ItemCount";
    fields[fields["ItemDestinationCategoryID"] = 791] = "ItemDestinationCategoryID";
    fields[fields["ItemDiscountAmount"] = 792] = "ItemDiscountAmount";
    fields[fields["ItemEndTime"] = 793] = "ItemEndTime";
    fields[fields["ItemID"] = 794] = "ItemID";
    fields[fields["ItemListLayout"] = 795] = "ItemListLayout";
    fields[fields["ItemListSortOrder"] = 796] = "ItemListSortOrder";
    fields[fields["ItemPolicyViolation"] = 797] = "ItemPolicyViolation";
    fields[fields["ItemPrice"] = 798] = "ItemPrice";
    fields[fields["ItemRatingDetails"] = 799] = "ItemRatingDetails";
    fields[fields["ItemReasonForFailure"] = 800] = "ItemReasonForFailure";
    fields[fields["ItemReceiptConfirmationToReceiveCount"] = 801] = "ItemReceiptConfirmationToReceiveCount";
    fields[fields["ItemReceiptToConfirmCount"] = 802] = "ItemReceiptToConfirmCount";
    fields[fields["ItemRevised"] = 803] = "ItemRevised";
    fields[fields["ItemsAvailableTo"] = 804] = "ItemsAvailableTo";
    fields[fields["ItemsAwaitingFeedback"] = 805] = "ItemsAwaitingFeedback";
    fields[fields["ItemsLocatedIn"] = 806] = "ItemsLocatedIn";
    fields[fields["ItemSort"] = 807] = "ItemSort";
    fields[fields["ItemSpecificDetails"] = 808] = "ItemSpecificDetails";
    fields[fields["ItemSpecifics"] = 809] = "ItemSpecifics";
    fields[fields["ItemSpecificsEnabled"] = 810] = "ItemSpecificsEnabled";
    fields[fields["ItemsPerPage"] = 811] = "ItemsPerPage";
    fields[fields["ItemStatus"] = 812] = "ItemStatus";
    fields[fields["ItemTitle"] = 813] = "ItemTitle";
    fields[fields["ItemToUnsubscribe"] = 814] = "ItemToUnsubscribe";
    fields[fields["ItemTransactionID"] = 815] = "ItemTransactionID";
    fields[fields["ItemTransactionIDArray"] = 816] = "ItemTransactionIDArray";
    fields[fields["ItemType"] = 817] = "ItemType";
    fields[fields["JurisdictionID"] = 818] = "JurisdictionID";
    fields[fields["JurisdictionName"] = 819] = "JurisdictionName";
    fields[fields["KeepMinActive"] = 820] = "KeepMinActive";
    fields[fields["KeepTokenValid"] = 821] = "KeepTokenValid";
    fields[fields["Keyword"] = 822] = "Keyword";
    fields[fields["LastAmountPaid"] = 823] = "LastAmountPaid";
    fields[fields["LastBidTime"] = 824] = "LastBidTime";
    fields[fields["LastEmailSentTime"] = 825] = "LastEmailSentTime";
    fields[fields["LastModifiedDate"] = 826] = "LastModifiedDate";
    fields[fields["LastModifiedTime"] = 827] = "LastModifiedTime";
    fields[fields["LastName"] = 828] = "LastName";
    fields[fields["LastOpenedTime"] = 829] = "LastOpenedTime";
    fields[fields["LastPaymentDate"] = 830] = "LastPaymentDate";
    fields[fields["LastTimeModified"] = 831] = "LastTimeModified";
    fields[fields["LastUpdateTime"] = 832] = "LastUpdateTime";
    fields[fields["LastUsedTime"] = 833] = "LastUsedTime";
    fields[fields["LayoutID"] = 834] = "LayoutID";
    fields[fields["LayoutTotal"] = 835] = "LayoutTotal";
    fields[fields["LeadCount"] = 836] = "LeadCount";
    fields[fields["LeafCategory"] = 837] = "LeafCategory";
    fields[fields["LeftNav"] = 838] = "LeftNav";
    fields[fields["LegalInvoice"] = 839] = "LegalInvoice";
    fields[fields["Level"] = 840] = "Level";
    fields[fields["LevelLimit"] = 841] = "LevelLimit";
    fields[fields["LineItem"] = 842] = "LineItem";
    fields[fields["LinkedPayPalAccount"] = 843] = "LinkedPayPalAccount";
    fields[fields["LinkID"] = 844] = "LinkID";
    fields[fields["LinkToInclude"] = 845] = "LinkToInclude";
    fields[fields["LinkType"] = 846] = "LinkType";
    fields[fields["ListAccordingToSchedule"] = 847] = "ListAccordingToSchedule";
    fields[fields["ListAtSpecificTimeOfDay"] = 848] = "ListAtSpecificTimeOfDay";
    fields[fields["ListedOn"] = 849] = "ListedOn";
    fields[fields["ListIfNoProduct"] = 850] = "ListIfNoProduct";
    fields[fields["ListingAnalyzerRecommendations"] = 851] = "ListingAnalyzerRecommendations";
    fields[fields["ListingCheckoutRedirectPreference"] = 852] = "ListingCheckoutRedirectPreference";
    fields[fields["ListingDesigner"] = 853] = "ListingDesigner";
    fields[fields["ListingDetails"] = 854] = "ListingDetails";
    fields[fields["ListingDuration"] = 855] = "ListingDuration";
    fields[fields["ListingDurations"] = 856] = "ListingDurations";
    fields[fields["ListingEnhancement"] = 857] = "ListingEnhancement";
    fields[fields["ListingFeatureDetails"] = 858] = "ListingFeatureDetails";
    fields[fields["ListingFlow"] = 859] = "ListingFlow";
    fields[fields["ListingHoldInventoryLevel"] = 860] = "ListingHoldInventoryLevel";
    fields[fields["ListingPeriodInWeeks"] = 861] = "ListingPeriodInWeeks";
    fields[fields["ListingRecommendations"] = 862] = "ListingRecommendations";
    fields[fields["ListingStartPriceDetails"] = 863] = "ListingStartPriceDetails";
    fields[fields["ListingStatus"] = 864] = "ListingStatus";
    fields[fields["ListingSubtype2"] = 865] = "ListingSubtype2";
    fields[fields["ListingTip"] = 866] = "ListingTip";
    fields[fields["ListingTipArray"] = 867] = "ListingTipArray";
    fields[fields["ListingTipFieldID"] = 868] = "ListingTipFieldID";
    fields[fields["ListingTipID"] = 869] = "ListingTipID";
    fields[fields["ListingTipMessageID"] = 870] = "ListingTipMessageID";
    fields[fields["ListingType"] = 871] = "ListingType";
    fields[fields["ListTimeFrom"] = 872] = "ListTimeFrom";
    fields[fields["ListTimeTo"] = 873] = "ListTimeTo";
    fields[fields["LocalListingDistance"] = 874] = "LocalListingDistance";
    fields[fields["LocalListingDistancesNonSubscription"] = 875] = "LocalListingDistancesNonSubscription";
    fields[fields["LocalListingDistancesRegular"] = 876] = "LocalListingDistancesRegular";
    fields[fields["LocalListingDistancesSpecialty"] = 877] = "LocalListingDistancesSpecialty";
    fields[fields["LocalMarketAdFormatEnabled"] = 878] = "LocalMarketAdFormatEnabled";
    fields[fields["LocalMarketAutoAcceptEnabled"] = 879] = "LocalMarketAutoAcceptEnabled";
    fields[fields["LocalMarketAutoDeclineEnabled"] = 880] = "LocalMarketAutoDeclineEnabled";
    fields[fields["LocalMarketBestOfferEnabled"] = 881] = "LocalMarketBestOfferEnabled";
    fields[fields["LocalMarketCompanyNameEnabled"] = 882] = "LocalMarketCompanyNameEnabled";
    fields[fields["LocalMarketContactByAddressEnabled"] = 883] = "LocalMarketContactByAddressEnabled";
    fields[fields["LocalMarketContactByEmailEnabled"] = 884] = "LocalMarketContactByEmailEnabled";
    fields[fields["LocalMarketContactByPhoneEnabled"] = 885] = "LocalMarketContactByPhoneEnabled";
    fields[fields["LocalMarketCounterOfferEnabled"] = 886] = "LocalMarketCounterOfferEnabled";
    fields[fields["LocalMarketNonSubscription"] = 887] = "LocalMarketNonSubscription";
    fields[fields["LocalMarketPaymentMethodCheckOutEnabled"] = 888] = "LocalMarketPaymentMethodCheckOutEnabled";
    fields[fields["LocalMarketPhoneCount"] = 889] = "LocalMarketPhoneCount";
    fields[fields["LocalMarketPremiumSubscription"] = 890] = "LocalMarketPremiumSubscription";
    fields[fields["LocalMarketRegularSubscription"] = 891] = "LocalMarketRegularSubscription";
    fields[fields["LocalMarketSellerContactDetailsEnabled"] = 892] = "LocalMarketSellerContactDetailsEnabled";
    fields[fields["LocalMarketShippingMethodEnabled"] = 893] = "LocalMarketShippingMethodEnabled";
    fields[fields["LocalMarketSpecialitySubscription"] = 894] = "LocalMarketSpecialitySubscription";
    fields[fields["LocalMarketStreetCount"] = 895] = "LocalMarketStreetCount";
    fields[fields["LocalPickup"] = 896] = "LocalPickup";
    fields[fields["Location"] = 897] = "Location";
    fields[fields["LocationDefaulted"] = 898] = "LocationDefaulted";
    fields[fields["Logo"] = 899] = "Logo";
    fields[fields["LogoArray"] = 900] = "LogoArray";
    fields[fields["LogoCustomized"] = 901] = "LogoCustomized";
    fields[fields["LogoID"] = 902] = "LogoID";
    fields[fields["LogoType"] = 903] = "LogoType";
    fields[fields["LogoURL"] = 904] = "LogoURL";
    fields[fields["LogoURLSelling"] = 905] = "LogoURLSelling";
    fields[fields["LongMessage"] = 906] = "LongMessage";
    fields[fields["LookupAttributeArray"] = 907] = "LookupAttributeArray";
    fields[fields["LostList"] = 908] = "LostList";
    fields[fields["LotSize"] = 909] = "LotSize";
    fields[fields["LSD"] = 910] = "LSD";
    fields[fields["MadeForOutletComparisonPrice"] = 911] = "MadeForOutletComparisonPrice";
    fields[fields["MailMessageType"] = 912] = "MailMessageType";
    fields[fields["MappedCategoryArray"] = 913] = "MappedCategoryArray";
    fields[fields["MappedDiscountProfileID"] = 914] = "MappedDiscountProfileID";
    fields[fields["MappedToShippingServiceID"] = 915] = "MappedToShippingServiceID";
    fields[fields["MarkUpMarkDownEvent"] = 916] = "MarkUpMarkDownEvent";
    fields[fields["MarkUpMarkDownHistory"] = 917] = "MarkUpMarkDownHistory";
    fields[fields["MatchType"] = 918] = "MatchType";
    fields[fields["MaxActiveItemCount"] = 919] = "MaxActiveItemCount";
    fields[fields["MaxBid"] = 920] = "MaxBid";
    fields[fields["MaxCategories"] = 921] = "MaxCategories";
    fields[fields["MaxCategoryLevels"] = 922] = "MaxCategoryLevels";
    fields[fields["MaxCharactersPerName"] = 923] = "MaxCharactersPerName";
    fields[fields["MaxCharactersPerValue"] = 924] = "MaxCharactersPerValue";
    fields[fields["MaxDepth"] = 925] = "MaxDepth";
    fields[fields["MaxDistance"] = 926] = "MaxDistance";
    fields[fields["MaxFlatShippingCost"] = 927] = "MaxFlatShippingCost";
    fields[fields["MaxFlatShippingCostCBTExempt"] = 928] = "MaxFlatShippingCostCBTExempt";
    fields[fields["MaxGranularFitmentCount"] = 929] = "MaxGranularFitmentCount";
    fields[fields["MaximumBuyerPolicyViolations"] = 930] = "MaximumBuyerPolicyViolations";
    fields[fields["MaximumItemCount"] = 931] = "MaximumItemCount";
    fields[fields["MaximumItemRequirements"] = 932] = "MaximumItemRequirements";
    fields[fields["MaximumQuantity"] = 933] = "MaximumQuantity";
    fields[fields["MaximumUnpaidItemStrikesCount"] = 934] = "MaximumUnpaidItemStrikesCount";
    fields[fields["MaximumUnpaidItemStrikesDuration"] = 935] = "MaximumUnpaidItemStrikesDuration";
    fields[fields["MaximumUnpaidItemStrikesInfo"] = 936] = "MaximumUnpaidItemStrikesInfo";
    fields[fields["MaxItemCompatibility"] = 937] = "MaxItemCompatibility";
    fields[fields["MaxItemSpecificsPerItem"] = 938] = "MaxItemSpecificsPerItem";
    fields[fields["MaxNames"] = 939] = "MaxNames";
    fields[fields["MaxNamesPerVariationSpecificsSet"] = 940] = "MaxNamesPerVariationSpecificsSet";
    fields[fields["MaxPrice"] = 941] = "MaxPrice";
    fields[fields["MaxResults"] = 942] = "MaxResults";
    fields[fields["MaxScheduledItems"] = 943] = "MaxScheduledItems";
    fields[fields["MaxScheduledMinutes"] = 944] = "MaxScheduledMinutes";
    fields[fields["MaxValues"] = 945] = "MaxValues";
    fields[fields["MaxValuesPerName"] = 946] = "MaxValuesPerName";
    fields[fields["MaxValuesPerVariationSpecificsSetName"] = 947] = "MaxValuesPerVariationSpecificsSetName";
    fields[fields["MaxVariationsPerItem"] = 948] = "MaxVariationsPerItem";
    fields[fields["MeasurementUnit"] = 949] = "MeasurementUnit";
    fields[fields["MechanicalCheckAccepted"] = 950] = "MechanicalCheckAccepted";
    fields[fields["MediaName"] = 951] = "MediaName";
    fields[fields["MediaURL"] = 952] = "MediaURL";
    fields[fields["MemberMessage"] = 953] = "MemberMessage";
    fields[fields["MemberMessageExchange"] = 954] = "MemberMessageExchange";
    fields[fields["MemberMessageID"] = 955] = "MemberMessageID";
    fields[fields["MemberURL"] = 956] = "MemberURL";
    fields[fields["Memo"] = 957] = "Memo";
    fields[fields["MerchandizingPref"] = 958] = "MerchandizingPref";
    fields[fields["MerchantPickupCode"] = 959] = "MerchantPickupCode";
    fields[fields["MerchDisplay"] = 960] = "MerchDisplay";
    fields[fields["Message"] = 961] = "Message";
    fields[fields["MessageCreationTime"] = 962] = "MessageCreationTime";
    fields[fields["MessageID"] = 963] = "MessageID";
    fields[fields["MessageIDs"] = 964] = "MessageIDs";
    fields[fields["MessageItem"] = 965] = "MessageItem";
    fields[fields["MessageMedia"] = 966] = "MessageMedia";
    fields[fields["Messages"] = 967] = "Messages";
    fields[fields["MessageSource"] = 968] = "MessageSource";
    fields[fields["MessageStatus"] = 969] = "MessageStatus";
    fields[fields["MessageStore"] = 970] = "MessageStore";
    fields[fields["MessageStoreCustomText"] = 971] = "MessageStoreCustomText";
    fields[fields["MessageText"] = 972] = "MessageText";
    fields[fields["MessageToSeller"] = 973] = "MessageToSeller";
    fields[fields["MessageType"] = 974] = "MessageType";
    fields[fields["MinActiveItemCount"] = 975] = "MinActiveItemCount";
    fields[fields["MinBuyItNowPricePercent"] = 976] = "MinBuyItNowPricePercent";
    fields[fields["MinimumAdvertisedPrice"] = 977] = "MinimumAdvertisedPrice";
    fields[fields["MinimumAdvertisedPriceExposure"] = 978] = "MinimumAdvertisedPriceExposure";
    fields[fields["MinimumBestOfferPrice"] = 979] = "MinimumBestOfferPrice";
    fields[fields["MinimumFeedbackScore"] = 980] = "MinimumFeedbackScore";
    fields[fields["MinimumRemnantSet"] = 981] = "MinimumRemnantSet";
    fields[fields["MinimumReservePrice"] = 982] = "MinimumReservePrice";
    fields[fields["MinimumToBid"] = 983] = "MinimumToBid";
    fields[fields["MinItemCompatibility"] = 984] = "MinItemCompatibility";
    fields[fields["MinPrice"] = 985] = "MinPrice";
    fields[fields["MinScheduledMinutes"] = 986] = "MinScheduledMinutes";
    fields[fields["MinValues"] = 987] = "MinValues";
    fields[fields["MismatchAmount"] = 988] = "MismatchAmount";
    fields[fields["MismatchType"] = 989] = "MismatchType";
    fields[fields["Mission"] = 990] = "Mission";
    fields[fields["ModifyActionCode"] = 991] = "ModifyActionCode";
    fields[fields["ModifyName"] = 992] = "ModifyName";
    fields[fields["ModifyNameList"] = 993] = "ModifyNameList";
    fields[fields["ModTime"] = 994] = "ModTime";
    fields[fields["ModTimeFrom"] = 995] = "ModTimeFrom";
    fields[fields["ModTimeTo"] = 996] = "ModTimeTo";
    fields[fields["MonetaryDetails"] = 997] = "MonetaryDetails";
    fields[fields["MotorsDealer"] = 998] = "MotorsDealer";
    fields[fields["MotorsDealerUsers"] = 999] = "MotorsDealerUsers";
    fields[fields["MotorsGermanySearchable"] = 1000] = "MotorsGermanySearchable";
    fields[fields["MotorVehicles"] = 1001] = "MotorVehicles";
    fields[fields["MPN"] = 1002] = "MPN";
    fields[fields["MultiLegShippingDetails"] = 1003] = "MultiLegShippingDetails";
    fields[fields["MultipleSellerPaymentID"] = 1004] = "MultipleSellerPaymentID";
    fields[fields["MyMaxBid"] = 1005] = "MyMaxBid";
    fields[fields["Name"] = 1006] = "Name";
    fields[fields["NameColor"] = 1007] = "NameColor";
    fields[fields["NameFace"] = 1008] = "NameFace";
    fields[fields["NameRecommendation"] = 1009] = "NameRecommendation";
    fields[fields["NameSize"] = 1010] = "NameSize";
    fields[fields["NameValueList"] = 1011] = "NameValueList";
    fields[fields["NegativeFeedbackLeftCount"] = 1012] = "NegativeFeedbackLeftCount";
    fields[fields["NegativeFeedbackPeriodArray"] = 1013] = "NegativeFeedbackPeriodArray";
    fields[fields["NegotiatedPrice"] = 1014] = "NegotiatedPrice";
    fields[fields["NetDetailAmount"] = 1015] = "NetDetailAmount";
    fields[fields["NetInsuranceFee"] = 1016] = "NetInsuranceFee";
    fields[fields["NetShippingFee"] = 1017] = "NetShippingFee";
    fields[fields["NetTotalAmount"] = 1018] = "NetTotalAmount";
    fields[fields["NeutralCommentCountFromSuspendedUsers"] = 1019] = "NeutralCommentCountFromSuspendedUsers";
    fields[fields["NeutralFeedbackLeftCount"] = 1020] = "NeutralFeedbackLeftCount";
    fields[fields["NeutralFeedbackPeriodArray"] = 1021] = "NeutralFeedbackPeriodArray";
    fields[fields["NewAlertCount"] = 1022] = "NewAlertCount";
    fields[fields["NewBestOffer"] = 1023] = "NewBestOffer";
    fields[fields["NewHighPriorityCount"] = 1024] = "NewHighPriorityCount";
    fields[fields["NewItemFilter"] = 1025] = "NewItemFilter";
    fields[fields["NewLeadCount"] = 1026] = "NewLeadCount";
    fields[fields["NewMessageCount"] = 1027] = "NewMessageCount";
    fields[fields["NewName"] = 1028] = "NewName";
    fields[fields["NewParentFolderID"] = 1029] = "NewParentFolderID";
    fields[fields["NewUser"] = 1030] = "NewUser";
    fields[fields["NextRetryTime"] = 1031] = "NextRetryTime";
    fields[fields["NonSubscription"] = 1032] = "NonSubscription";
    fields[fields["Note"] = 1033] = "Note";
    fields[fields["Notes"] = 1034] = "Notes";
    fields[fields["NotesFromBuyer"] = 1035] = "NotesFromBuyer";
    fields[fields["NotesToBuyer"] = 1036] = "NotesToBuyer";
    fields[fields["NotesToSeller"] = 1037] = "NotesToSeller";
    fields[fields["NoteText"] = 1038] = "NoteText";
    fields[fields["NotificationDetails"] = 1039] = "NotificationDetails";
    fields[fields["NotificationDetailsArray"] = 1040] = "NotificationDetailsArray";
    fields[fields["NotificationEnable"] = 1041] = "NotificationEnable";
    fields[fields["NotificationPayloadType"] = 1042] = "NotificationPayloadType";
    fields[fields["NotificationStatistics"] = 1043] = "NotificationStatistics";
    fields[fields["NumberOfDays"] = 1044] = "NumberOfDays";
    fields[fields["NumberOfPolicyViolations"] = 1045] = "NumberOfPolicyViolations";
    fields[fields["NumOfReqSellerActions"] = 1046] = "NumOfReqSellerActions";
    fields[fields["ObsoleteLayoutID"] = 1047] = "ObsoleteLayoutID";
    fields[fields["ObsoleteThemeID"] = 1048] = "ObsoleteThemeID";
    fields[fields["Offer"] = 1049] = "Offer";
    fields[fields["OfferGlobalShippingProgramPreference"] = 1050] = "OfferGlobalShippingProgramPreference";
    fields[fields["OnVacation"] = 1051] = "OnVacation";
    fields[fields["Operation"] = 1052] = "Operation";
    fields[fields["OptedIn"] = 1053] = "OptedIn";
    fields[fields["OptimalPictureSize"] = 1054] = "OptimalPictureSize";
    fields[fields["OptInStatus"] = 1055] = "OptInStatus";
    fields[fields["Order"] = 1056] = "Order";
    fields[fields["OrderAmount"] = 1057] = "OrderAmount";
    fields[fields["OrderArray"] = 1058] = "OrderArray";
    fields[fields["OrderHandlingAmount"] = 1059] = "OrderHandlingAmount";
    fields[fields["OrderID"] = 1060] = "OrderID";
    fields[fields["OrderIDArray"] = 1061] = "OrderIDArray";
    fields[fields["OrderLineItemID"] = 1062] = "OrderLineItemID";
    fields[fields["OrderRole"] = 1063] = "OrderRole";
    fields[fields["OrdersPerPage"] = 1064] = "OrdersPerPage";
    fields[fields["OrderStatus"] = 1065] = "OrderStatus";
    fields[fields["OrderStatusFilter"] = 1066] = "OrderStatusFilter";
    fields[fields["OrderTransaction"] = 1067] = "OrderTransaction";
    fields[fields["OrderTransactionArray"] = 1068] = "OrderTransactionArray";
    fields[fields["OriginalItemPrice"] = 1069] = "OriginalItemPrice";
    fields[fields["OriginalItemShippingCost"] = 1070] = "OriginalItemShippingCost";
    fields[fields["OriginalPrice"] = 1071] = "OriginalPrice";
    fields[fields["OriginalRetailPrice"] = 1072] = "OriginalRetailPrice";
    fields[fields["OriginatingPostalCode"] = 1073] = "OriginatingPostalCode";
    fields[fields["ORPA"] = 1074] = "ORPA";
    fields[fields["ORRA"] = 1075] = "ORRA";
    fields[fields["OtherPartyName"] = 1076] = "OtherPartyName";
    fields[fields["OtherPartyRole"] = 1077] = "OtherPartyRole";
    fields[fields["OutbidCount"] = 1078] = "OutbidCount";
    fields[fields["OutOfStockControl"] = 1079] = "OutOfStockControl";
    fields[fields["OutputSelector"] = 1080] = "OutputSelector";
    fields[fields["OverrideGSPserviceWithIntlService"] = 1081] = "OverrideGSPserviceWithIntlService";
    fields[fields["OverrideGSPServiceWithIntlServicePreference"] = 1082] = "OverrideGSPServiceWithIntlServicePreference";
    fields[fields["PackageDepth"] = 1083] = "PackageDepth";
    fields[fields["PackageID"] = 1084] = "PackageID";
    fields[fields["PackageLength"] = 1085] = "PackageLength";
    fields[fields["PackageWidth"] = 1086] = "PackageWidth";
    fields[fields["PackagingHandlingCosts"] = 1087] = "PackagingHandlingCosts";
    fields[fields["PageID"] = 1088] = "PageID";
    fields[fields["PageNumber"] = 1089] = "PageNumber";
    fields[fields["Pagination"] = 1090] = "Pagination";
    fields[fields["PaginationResult"] = 1091] = "PaginationResult";
    fields[fields["Paid"] = 1092] = "Paid";
    fields[fields["PaidStatus"] = 1093] = "PaidStatus";
    fields[fields["PaidTime"] = 1094] = "PaidTime";
    fields[fields["PaisaPayAlert"] = 1095] = "PaisaPayAlert";
    fields[fields["PaisaPayEscrowEMIStatus"] = 1096] = "PaisaPayEscrowEMIStatus";
    fields[fields["PaisaPayFullEscrowEnabled"] = 1097] = "PaisaPayFullEscrowEnabled";
    fields[fields["PaisaPayID"] = 1098] = "PaisaPayID";
    fields[fields["PaisaPayStatus"] = 1099] = "PaisaPayStatus";
    fields[fields["ParentFolderID"] = 1100] = "ParentFolderID";
    fields[fields["ParentMessageID"] = 1101] = "ParentMessageID";
    fields[fields["ParentName"] = 1102] = "ParentName";
    fields[fields["ParentValue"] = 1103] = "ParentValue";
    fields[fields["PastDue"] = 1104] = "PastDue";
    fields[fields["Patent"] = 1105] = "Patent";
    fields[fields["Payee"] = 1106] = "Payee";
    fields[fields["Payer"] = 1107] = "Payer";
    fields[fields["PayloadVersion"] = 1108] = "PayloadVersion";
    fields[fields["Payment"] = 1109] = "Payment";
    fields[fields["PaymentAllowedSite"] = 1110] = "PaymentAllowedSite";
    fields[fields["PaymentAmount"] = 1111] = "PaymentAmount";
    fields[fields["PaymentDetails"] = 1112] = "PaymentDetails";
    fields[fields["PaymentEdited"] = 1113] = "PaymentEdited";
    fields[fields["PaymentHoldDetails"] = 1114] = "PaymentHoldDetails";
    fields[fields["PaymentHoldReason"] = 1115] = "PaymentHoldReason";
    fields[fields["PaymentHoldStatus"] = 1116] = "PaymentHoldStatus";
    fields[fields["PaymentInstructions"] = 1117] = "PaymentInstructions";
    fields[fields["PaymentMethod"] = 1118] = "PaymentMethod";
    fields[fields["PaymentMethods"] = 1119] = "PaymentMethods";
    fields[fields["PaymentMethodUsed"] = 1120] = "PaymentMethodUsed";
    fields[fields["PaymentOptionDetails"] = 1121] = "PaymentOptionDetails";
    fields[fields["PaymentOptionsGroup"] = 1122] = "PaymentOptionsGroup";
    fields[fields["PaymentOrRefundAmount"] = 1123] = "PaymentOrRefundAmount";
    fields[fields["PaymentProfileCategoryGroup"] = 1124] = "PaymentProfileCategoryGroup";
    fields[fields["PaymentProfileID"] = 1125] = "PaymentProfileID";
    fields[fields["PaymentProfileName"] = 1126] = "PaymentProfileName";
    fields[fields["PaymentReferenceID"] = 1127] = "PaymentReferenceID";
    fields[fields["Payments"] = 1128] = "Payments";
    fields[fields["PaymentsPerPage"] = 1129] = "PaymentsPerPage";
    fields[fields["PaymentStatus"] = 1130] = "PaymentStatus";
    fields[fields["PaymentTime"] = 1131] = "PaymentTime";
    fields[fields["PaymentTimeFrom"] = 1132] = "PaymentTimeFrom";
    fields[fields["PaymentTimeTo"] = 1133] = "PaymentTimeTo";
    fields[fields["PaymentToReceiveCount"] = 1134] = "PaymentToReceiveCount";
    fields[fields["PaymentToSendCount"] = 1135] = "PaymentToSendCount";
    fields[fields["PaymentType"] = 1136] = "PaymentType";
    fields[fields["PayoutMethod"] = 1137] = "PayoutMethod";
    fields[fields["PayoutMethodSet"] = 1138] = "PayoutMethodSet";
    fields[fields["PayPalAccountLevel"] = 1139] = "PayPalAccountLevel";
    fields[fields["PayPalAccountStatus"] = 1140] = "PayPalAccountStatus";
    fields[fields["PayPalAccountType"] = 1141] = "PayPalAccountType";
    fields[fields["PayPalAlwaysOn"] = 1142] = "PayPalAlwaysOn";
    fields[fields["PayPalBuyerProtectionEnabled"] = 1143] = "PayPalBuyerProtectionEnabled";
    fields[fields["PayPalEmailAddress"] = 1144] = "PayPalEmailAddress";
    fields[fields["PayPalPreferred"] = 1145] = "PayPalPreferred";
    fields[fields["PayPalRequired"] = 1146] = "PayPalRequired";
    fields[fields["PayPalRequiredForStoreOwner"] = 1147] = "PayPalRequiredForStoreOwner";
    fields[fields["PayPalTransactionID"] = 1148] = "PayPalTransactionID";
    fields[fields["PendingRTERequestCount"] = 1149] = "PendingRTERequestCount";
    fields[fields["Percent"] = 1150] = "Percent";
    fields[fields["PercentItemFound"] = 1151] = "PercentItemFound";
    fields[fields["Performance"] = 1152] = "Performance";
    fields[fields["Period"] = 1153] = "Period";
    fields[fields["PeriodicHardLimit"] = 1154] = "PeriodicHardLimit";
    fields[fields["PeriodicSoftLimit"] = 1155] = "PeriodicSoftLimit";
    fields[fields["PeriodicStartDate"] = 1156] = "PeriodicStartDate";
    fields[fields["PeriodicUsage"] = 1157] = "PeriodicUsage";
    fields[fields["PeriodInDays"] = 1158] = "PeriodInDays";
    fields[fields["Phone"] = 1159] = "Phone";
    fields[fields["Phone2"] = 1160] = "Phone2";
    fields[fields["Phone2AreaOrCityCode"] = 1161] = "Phone2AreaOrCityCode";
    fields[fields["Phone2CountryCode"] = 1162] = "Phone2CountryCode";
    fields[fields["Phone2CountryPrefix"] = 1163] = "Phone2CountryPrefix";
    fields[fields["Phone2LocalNumber"] = 1164] = "Phone2LocalNumber";
    fields[fields["PhoneAreaOrCityCode"] = 1165] = "PhoneAreaOrCityCode";
    fields[fields["PhoneCountryCode"] = 1166] = "PhoneCountryCode";
    fields[fields["PhoneCountryPrefix"] = 1167] = "PhoneCountryPrefix";
    fields[fields["PhoneLocalNumber"] = 1168] = "PhoneLocalNumber";
    fields[fields["PhotoDisplay"] = 1169] = "PhotoDisplay";
    fields[fields["PickupDetails"] = 1170] = "PickupDetails";
    fields[fields["PickupFulfillmentTime"] = 1171] = "PickupFulfillmentTime";
    fields[fields["PickupInStoreDetails"] = 1172] = "PickupInStoreDetails";
    fields[fields["PickupMethod"] = 1173] = "PickupMethod";
    fields[fields["PickupMethodSelected"] = 1174] = "PickupMethodSelected";
    fields[fields["PickupOptions"] = 1175] = "PickupOptions";
    fields[fields["PickupPriority"] = 1176] = "PickupPriority";
    fields[fields["PickupStatus"] = 1177] = "PickupStatus";
    fields[fields["PickupStoreID"] = 1178] = "PickupStoreID";
    fields[fields["PictureData"] = 1179] = "PictureData";
    fields[fields["PictureDetails"] = 1180] = "PictureDetails";
    fields[fields["PictureFormat"] = 1181] = "PictureFormat";
    fields[fields["PictureHeight"] = 1182] = "PictureHeight";
    fields[fields["PictureName"] = 1183] = "PictureName";
    fields[fields["Pictures"] = 1184] = "Pictures";
    fields[fields["PictureSet"] = 1185] = "PictureSet";
    fields[fields["PictureSetMember"] = 1186] = "PictureSetMember";
    fields[fields["PictureSource"] = 1187] = "PictureSource";
    fields[fields["PictureSystemVersion"] = 1188] = "PictureSystemVersion";
    fields[fields["PictureUploadPolicy"] = 1189] = "PictureUploadPolicy";
    fields[fields["PictureURL"] = 1190] = "PictureURL";
    fields[fields["PictureWatermark"] = 1191] = "PictureWatermark";
    fields[fields["PictureWidth"] = 1192] = "PictureWidth";
    fields[fields["Platform"] = 1193] = "Platform";
    fields[fields["PolicyID"] = 1194] = "PolicyID";
    fields[fields["PolicyText"] = 1195] = "PolicyText";
    fields[fields["PolicyViolationDuration"] = 1196] = "PolicyViolationDuration";
    fields[fields["PositiveFeedbackLeftCount"] = 1197] = "PositiveFeedbackLeftCount";
    fields[fields["PositiveFeedbackPercent"] = 1198] = "PositiveFeedbackPercent";
    fields[fields["PositiveFeedbackPeriodArray"] = 1199] = "PositiveFeedbackPeriodArray";
    fields[fields["PostalCode"] = 1200] = "PostalCode";
    fields[fields["PostCheckoutExperienceEnabled"] = 1201] = "PostCheckoutExperienceEnabled";
    fields[fields["PowerSellerStatus"] = 1202] = "PowerSellerStatus";
    fields[fields["PreferenceLevel"] = 1203] = "PreferenceLevel";
    fields[fields["PreferredLocation"] = 1204] = "PreferredLocation";
    fields[fields["PremiumSubscription"] = 1205] = "PremiumSubscription";
    fields[fields["PreviewEnabled"] = 1206] = "PreviewEnabled";
    fields[fields["Price"] = 1207] = "Price";
    fields[fields["PriceMax"] = 1208] = "PriceMax";
    fields[fields["PriceMin"] = 1209] = "PriceMin";
    fields[fields["PricingRecommendations"] = 1210] = "PricingRecommendations";
    fields[fields["PricingTreatment"] = 1211] = "PricingTreatment";
    fields[fields["Primary"] = 1212] = "Primary";
    fields[fields["PrimaryCategory"] = 1213] = "PrimaryCategory";
    fields[fields["PrimaryCategoryID"] = 1214] = "PrimaryCategoryID";
    fields[fields["PrimaryScheme"] = 1215] = "PrimaryScheme";
    fields[fields["PrintedDate"] = 1216] = "PrintedDate";
    fields[fields["PrintedTime"] = 1217] = "PrintedTime";
    fields[fields["Priority"] = 1218] = "Priority";
    fields[fields["PrivateListing"] = 1219] = "PrivateListing";
    fields[fields["PrivateNotes"] = 1220] = "PrivateNotes";
    fields[fields["Product"] = 1221] = "Product";
    fields[fields["ProductCreationEnabled"] = 1222] = "ProductCreationEnabled";
    fields[fields["ProductDetailsURL"] = 1223] = "ProductDetailsURL";
    fields[fields["ProductFinderBuySide"] = 1224] = "ProductFinderBuySide";
    fields[fields["ProductFinderID"] = 1225] = "ProductFinderID";
    fields[fields["ProductFinderIDs"] = 1226] = "ProductFinderIDs";
    fields[fields["ProductID"] = 1227] = "ProductID";
    fields[fields["ProductInfo"] = 1228] = "ProductInfo";
    fields[fields["ProductListingDetails"] = 1229] = "ProductListingDetails";
    fields[fields["ProductName"] = 1230] = "ProductName";
    fields[fields["ProductRecommendations"] = 1231] = "ProductRecommendations";
    fields[fields["ProductReferenceID"] = 1232] = "ProductReferenceID";
    fields[fields["ProductSearchPageAvailable"] = 1233] = "ProductSearchPageAvailable";
    fields[fields["ProductState"] = 1234] = "ProductState";
    fields[fields["ProductSuggestion"] = 1235] = "ProductSuggestion";
    fields[fields["ProductSuggestions"] = 1236] = "ProductSuggestions";
    fields[fields["ProfileID"] = 1237] = "ProfileID";
    fields[fields["ProfileName"] = 1238] = "ProfileName";
    fields[fields["ProfileType"] = 1239] = "ProfileType";
    fields[fields["ProfitPercent"] = 1240] = "ProfitPercent";
    fields[fields["PromotedItem"] = 1241] = "PromotedItem";
    fields[fields["PromotionalDiscount"] = 1242] = "PromotionalDiscount";
    fields[fields["PromotionalSale"] = 1243] = "PromotionalSale";
    fields[fields["PromotionalSaleDetails"] = 1244] = "PromotionalSaleDetails";
    fields[fields["PromotionalSaleEndTime"] = 1245] = "PromotionalSaleEndTime";
    fields[fields["PromotionalSaleID"] = 1246] = "PromotionalSaleID";
    fields[fields["PromotionalSaleItemIDArray"] = 1247] = "PromotionalSaleItemIDArray";
    fields[fields["PromotionalSaleName"] = 1248] = "PromotionalSaleName";
    fields[fields["PromotionalSaleStartTime"] = 1249] = "PromotionalSaleStartTime";
    fields[fields["PromotionalSaleStatus"] = 1250] = "PromotionalSaleStatus";
    fields[fields["PromotionalSaleType"] = 1251] = "PromotionalSaleType";
    fields[fields["PromotionalShippingDiscount"] = 1252] = "PromotionalShippingDiscount";
    fields[fields["PromotionalShippingDiscountDetails"] = 1253] = "PromotionalShippingDiscountDetails";
    fields[fields["PromotionMethod"] = 1254] = "PromotionMethod";
    fields[fields["PromotionRuleArray"] = 1255] = "PromotionRuleArray";
    fields[fields["PromotionViewMode"] = 1256] = "PromotionViewMode";
    fields[fields["ProPack"] = 1257] = "ProPack";
    fields[fields["ProPackEnabled"] = 1258] = "ProPackEnabled";
    fields[fields["ProPackPlusEnabled"] = 1259] = "ProPackPlusEnabled";
    fields[fields["ProStoresPreference"] = 1260] = "ProStoresPreference";
    fields[fields["ProStoresStoreName"] = 1261] = "ProStoresStoreName";
    fields[fields["ProxyItem"] = 1262] = "ProxyItem";
    fields[fields["PurchaseProtection"] = 1263] = "PurchaseProtection";
    fields[fields["PurchaseReminderEmailPreferences"] = 1264] = "PurchaseReminderEmailPreferences";
    fields[fields["PurchaseTimeFrame"] = 1265] = "PurchaseTimeFrame";
    fields[fields["QualifiedForAuctionOneDayDuration"] = 1266] = "QualifiedForAuctionOneDayDuration";
    fields[fields["QualifiedForFixedPriceOneDayDuration"] = 1267] = "QualifiedForFixedPriceOneDayDuration";
    fields[fields["QualifiesForB2BVAT"] = 1268] = "QualifiesForB2BVAT";
    fields[fields["QualifiesForBuyItNow"] = 1269] = "QualifiesForBuyItNow";
    fields[fields["QualifiesForBuyItNowMultiple"] = 1270] = "QualifiesForBuyItNowMultiple";
    fields[fields["QualifiesForSelling"] = 1271] = "QualifiesForSelling";
    fields[fields["QualifiesForVariations"] = 1272] = "QualifiesForVariations";
    fields[fields["Quantity"] = 1273] = "Quantity";
    fields[fields["QuantityActive"] = 1274] = "QuantityActive";
    fields[fields["QuantityAvailable"] = 1275] = "QuantityAvailable";
    fields[fields["QuantityAvailableHint"] = 1276] = "QuantityAvailableHint";
    fields[fields["QuantityBid"] = 1277] = "QuantityBid";
    fields[fields["QuantityInfo"] = 1278] = "QuantityInfo";
    fields[fields["QuantityLimitRemaining"] = 1279] = "QuantityLimitRemaining";
    fields[fields["QuantityOperator"] = 1280] = "QuantityOperator";
    fields[fields["QuantityPurchased"] = 1281] = "QuantityPurchased";
    fields[fields["QuantityRestrictionPerBuyer"] = 1282] = "QuantityRestrictionPerBuyer";
    fields[fields["QuantityScheduled"] = 1283] = "QuantityScheduled";
    fields[fields["QuantitySold"] = 1284] = "QuantitySold";
    fields[fields["QuantitySoldByPickupInStore"] = 1285] = "QuantitySoldByPickupInStore";
    fields[fields["QuantityThreshold"] = 1286] = "QuantityThreshold";
    fields[fields["QuantityUnsold"] = 1287] = "QuantityUnsold";
    fields[fields["QuantityWon"] = 1288] = "QuantityWon";
    fields[fields["Query"] = 1289] = "Query";
    fields[fields["QueryKeywords"] = 1290] = "QueryKeywords";
    fields[fields["Question"] = 1291] = "Question";
    fields[fields["QuestionCount"] = 1292] = "QuestionCount";
    fields[fields["QuestionType"] = 1293] = "QuestionType";
    fields[fields["QueuedNewCount"] = 1294] = "QueuedNewCount";
    fields[fields["QueuedPendingCount"] = 1295] = "QueuedPendingCount";
    fields[fields["Ranking"] = 1296] = "Ranking";
    fields[fields["RateTableDetails"] = 1297] = "RateTableDetails";
    fields[fields["Rating"] = 1298] = "Rating";
    fields[fields["RatingCount"] = 1299] = "RatingCount";
    fields[fields["RatingDetail"] = 1300] = "RatingDetail";
    fields[fields["Read"] = 1301] = "Read";
    fields[fields["Reason"] = 1302] = "Reason";
    fields[fields["ReasonCodeDetail"] = 1303] = "ReasonCodeDetail";
    fields[fields["ReasonCodeID"] = 1304] = "ReasonCodeID";
    fields[fields["ReasonHideFromSearch"] = 1305] = "ReasonHideFromSearch";
    fields[fields["ReceiveDate"] = 1306] = "ReceiveDate";
    fields[fields["ReceivedTopRatedDiscount"] = 1307] = "ReceivedTopRatedDiscount";
    fields[fields["RecipientBidderUserID"] = 1308] = "RecipientBidderUserID";
    fields[fields["RecipientID"] = 1309] = "RecipientID";
    fields[fields["RecipientPostalCode"] = 1310] = "RecipientPostalCode";
    fields[fields["RecipientRelationType"] = 1311] = "RecipientRelationType";
    fields[fields["RecipientUserID"] = 1312] = "RecipientUserID";
    fields[fields["Recommendation"] = 1313] = "Recommendation";
    fields[fields["RecommendationEngine"] = 1314] = "RecommendationEngine";
    fields[fields["Recommendations"] = 1315] = "Recommendations";
    fields[fields["Recommended"] = 1316] = "Recommended";
    fields[fields["RecoupmentPolicyConsent"] = 1317] = "RecoupmentPolicyConsent";
    fields[fields["RecoupmentPolicyDetails"] = 1318] = "RecoupmentPolicyDetails";
    fields[fields["ReduceReserveAllowed"] = 1319] = "ReduceReserveAllowed";
    fields[fields["ReferenceID"] = 1320] = "ReferenceID";
    fields[fields["RefNumber"] = 1321] = "RefNumber";
    fields[fields["Refund"] = 1322] = "Refund";
    fields[fields["RefundAmount"] = 1323] = "RefundAmount";
    fields[fields["RefundArray"] = 1324] = "RefundArray";
    fields[fields["RefundCancelledCount"] = 1325] = "RefundCancelledCount";
    fields[fields["RefundCompletionTime"] = 1326] = "RefundCompletionTime";
    fields[fields["RefundFailureCode"] = 1327] = "RefundFailureCode";
    fields[fields["RefundFailureReason"] = 1328] = "RefundFailureReason";
    fields[fields["RefundFromSeller"] = 1329] = "RefundFromSeller";
    fields[fields["RefundFundingSource"] = 1330] = "RefundFundingSource";
    fields[fields["RefundFundingSourceArray"] = 1331] = "RefundFundingSourceArray";
    fields[fields["RefundID"] = 1332] = "RefundID";
    fields[fields["RefundingSourceType"] = 1333] = "RefundingSourceType";
    fields[fields["RefundInitiatedCount"] = 1334] = "RefundInitiatedCount";
    fields[fields["RefundLine"] = 1335] = "RefundLine";
    fields[fields["RefundLineArray"] = 1336] = "RefundLineArray";
    fields[fields["RefundMessage"] = 1337] = "RefundMessage";
    fields[fields["RefundOnHoldCount"] = 1338] = "RefundOnHoldCount";
    fields[fields["RefundOption"] = 1339] = "RefundOption";
    fields[fields["RefundReason"] = 1340] = "RefundReason";
    fields[fields["RefundRequestedTime"] = 1341] = "RefundRequestedTime";
    fields[fields["Refunds"] = 1342] = "Refunds";
    fields[fields["RefundStatus"] = 1343] = "RefundStatus";
    fields[fields["RefundTime"] = 1344] = "RefundTime";
    fields[fields["RefundTo"] = 1345] = "RefundTo";
    fields[fields["RefundTransaction"] = 1346] = "RefundTransaction";
    fields[fields["RefundTransactionArray"] = 1347] = "RefundTransactionArray";
    fields[fields["RefundType"] = 1348] = "RefundType";
    fields[fields["Region"] = 1349] = "Region";
    fields[fields["RegionDetails"] = 1350] = "RegionDetails";
    fields[fields["RegionOfOriginDetails"] = 1351] = "RegionOfOriginDetails";
    fields[fields["RegisteredBusinessSeller"] = 1352] = "RegisteredBusinessSeller";
    fields[fields["RegistrationAddress"] = 1353] = "RegistrationAddress";
    fields[fields["RegistrationDate"] = 1354] = "RegistrationDate";
    fields[fields["RegularSubscription"] = 1355] = "RegularSubscription";
    fields[fields["Relationship"] = 1356] = "Relationship";
    fields[fields["RelistAfterDays"] = 1357] = "RelistAfterDays";
    fields[fields["RelistAfterHours"] = 1358] = "RelistAfterHours";
    fields[fields["RelistAtSpecificTimeOfDay"] = 1359] = "RelistAtSpecificTimeOfDay";
    fields[fields["RelistCondition"] = 1360] = "RelistCondition";
    fields[fields["Relisted"] = 1361] = "Relisted";
    fields[fields["RelistedItemID"] = 1362] = "RelistedItemID";
    fields[fields["RelistingNeededCount"] = 1363] = "RelistingNeededCount";
    fields[fields["RelistLink"] = 1364] = "RelistLink";
    fields[fields["RelistParentID"] = 1365] = "RelistParentID";
    fields[fields["RemoveAllExcludedUsers"] = 1366] = "RemoveAllExcludedUsers";
    fields[fields["RemoveAllItems"] = 1367] = "RemoveAllItems";
    fields[fields["RepeatBuyerCount"] = 1368] = "RepeatBuyerCount";
    fields[fields["RepeatBuyerPercent"] = 1369] = "RepeatBuyerPercent";
    fields[fields["ReplaceAll"] = 1370] = "ReplaceAll";
    fields[fields["Replied"] = 1371] = "Replied";
    fields[fields["ReportedItem"] = 1372] = "ReportedItem";
    fields[fields["ReportedItemDetails"] = 1373] = "ReportedItemDetails";
    fields[fields["ReportItem"] = 1374] = "ReportItem";
    fields[fields["ReportItems"] = 1375] = "ReportItems";
    fields[fields["RequesterID"] = 1376] = "RequesterID";
    fields[fields["RequiredSellerAction"] = 1377] = "RequiredSellerAction";
    fields[fields["RequiredSellerActionArray"] = 1378] = "RequiredSellerActionArray";
    fields[fields["RequiredShipPhoneNumberPreference"] = 1379] = "RequiredShipPhoneNumberPreference";
    fields[fields["ReserveLowered"] = 1380] = "ReserveLowered";
    fields[fields["ReserveMet"] = 1381] = "ReserveMet";
    fields[fields["ReservePrice"] = 1382] = "ReservePrice";
    fields[fields["ReservePriceAllowed"] = 1383] = "ReservePriceAllowed";
    fields[fields["ReserveRemoved"] = 1384] = "ReserveRemoved";
    fields[fields["ResetDefaultSubjects"] = 1385] = "ResetDefaultSubjects";
    fields[fields["ResolutionTime"] = 1386] = "ResolutionTime";
    fields[fields["RespondToBestOffer"] = 1387] = "RespondToBestOffer";
    fields[fields["Response"] = 1388] = "Response";
    fields[fields["ResponseDetails"] = 1389] = "ResponseDetails";
    fields[fields["ResponseEnabled"] = 1390] = "ResponseEnabled";
    fields[fields["ResponseReplaced"] = 1391] = "ResponseReplaced";
    fields[fields["ResponseText"] = 1392] = "ResponseText";
    fields[fields["ResponseType"] = 1393] = "ResponseType";
    fields[fields["ResponseURL"] = 1394] = "ResponseURL";
    fields[fields["RestockAlert"] = 1395] = "RestockAlert";
    fields[fields["RestockingFeeValue"] = 1396] = "RestockingFeeValue";
    fields[fields["RestockingFeeValueOption"] = 1397] = "RestockingFeeValueOption";
    fields[fields["RestockThreshold"] = 1398] = "RestockThreshold";
    fields[fields["RestrictedToBusiness"] = 1399] = "RestrictedToBusiness";
    fields[fields["RESTToken"] = 1400] = "RESTToken";
    fields[fields["Retries"] = 1401] = "Retries";
    fields[fields["ReturnAllSites"] = 1402] = "ReturnAllSites";
    fields[fields["ReturnDate"] = 1403] = "ReturnDate";
    fields[fields["ReturnedItemCountActual"] = 1404] = "ReturnedItemCountActual";
    fields[fields["ReturnedOrderCountActual"] = 1405] = "ReturnedOrderCountActual";
    fields[fields["ReturnedPaymentCountActual"] = 1406] = "ReturnedPaymentCountActual";
    fields[fields["ReturnedTransactionCountActual"] = 1407] = "ReturnedTransactionCountActual";
    fields[fields["ReturnPolicy"] = 1408] = "ReturnPolicy";
    fields[fields["ReturnPolicyDetails"] = 1409] = "ReturnPolicyDetails";
    fields[fields["ReturnPolicyEnabled"] = 1410] = "ReturnPolicyEnabled";
    fields[fields["ReturnPolicyProfileCategoryGroup"] = 1411] = "ReturnPolicyProfileCategoryGroup";
    fields[fields["ReturnProfileID"] = 1412] = "ReturnProfileID";
    fields[fields["ReturnProfileName"] = 1413] = "ReturnProfileName";
    fields[fields["ReturnsAccepted"] = 1414] = "ReturnsAccepted";
    fields[fields["ReturnsAcceptedOption"] = 1415] = "ReturnsAcceptedOption";
    fields[fields["ReturnSearchResultOnDuplicates"] = 1416] = "ReturnSearchResultOnDuplicates";
    fields[fields["ReturnsWithin"] = 1417] = "ReturnsWithin";
    fields[fields["ReturnsWithinOption"] = 1418] = "ReturnsWithinOption";
    fields[fields["RevisePriceAllowed"] = 1419] = "RevisePriceAllowed";
    fields[fields["ReviseQuantityAllowed"] = 1420] = "ReviseQuantityAllowed";
    fields[fields["ReviseStatus"] = 1421] = "ReviseStatus";
    fields[fields["RevocationTime"] = 1422] = "RevocationTime";
    fields[fields["RightsOwnerID"] = 1423] = "RightsOwnerID";
    fields[fields["Role"] = 1424] = "Role";
    fields[fields["RootCategoryID"] = 1425] = "RootCategoryID";
    fields[fields["RTEToProcessCount"] = 1426] = "RTEToProcessCount";
    fields[fields["RuleCurrentStatus"] = 1427] = "RuleCurrentStatus";
    fields[fields["RuleStatus"] = 1428] = "RuleStatus";
    fields[fields["RuName"] = 1429] = "RuName";
    fields[fields["SafePaymentExempt"] = 1430] = "SafePaymentExempt";
    fields[fields["SafePaymentRequired"] = 1431] = "SafePaymentRequired";
    fields[fields["SaleDateRange"] = 1432] = "SaleDateRange";
    fields[fields["SalePrice"] = 1433] = "SalePrice";
    fields[fields["SaleRecord"] = 1434] = "SaleRecord";
    fields[fields["SaleRecordID"] = 1435] = "SaleRecordID";
    fields[fields["SalesTax"] = 1436] = "SalesTax";
    fields[fields["SalesTaxAmount"] = 1437] = "SalesTaxAmount";
    fields[fields["SalesTaxPercent"] = 1438] = "SalesTaxPercent";
    fields[fields["SalesTaxState"] = 1439] = "SalesTaxState";
    fields[fields["SaleTemplateGroupID"] = 1440] = "SaleTemplateGroupID";
    fields[fields["SaleTemplateID"] = 1441] = "SaleTemplateID";
    fields[fields["SaleTemplateName"] = 1442] = "SaleTemplateName";
    fields[fields["ScheduledDeliveryTimeMax"] = 1443] = "ScheduledDeliveryTimeMax";
    fields[fields["ScheduledDeliveryTimeMin"] = 1444] = "ScheduledDeliveryTimeMin";
    fields[fields["ScheduledList"] = 1445] = "ScheduledList";
    fields[fields["ScheduleTime"] = 1446] = "ScheduleTime";
    fields[fields["SchedulingInfo"] = 1447] = "SchedulingInfo";
    fields[fields["Search"] = 1448] = "Search";
    fields[fields["SearchBox"] = 1449] = "SearchBox";
    fields[fields["SearchFlag"] = 1450] = "SearchFlag";
    fields[fields["SearchKeywords"] = 1451] = "SearchKeywords";
    fields[fields["SearchName"] = 1452] = "SearchName";
    fields[fields["SearchQuery"] = 1453] = "SearchQuery";
    fields[fields["SearchSortOrder"] = 1454] = "SearchSortOrder";
    fields[fields["SearchStanding"] = 1455] = "SearchStanding";
    fields[fields["SearchType"] = 1456] = "SearchType";
    fields[fields["SearchValue"] = 1457] = "SearchValue";
    fields[fields["Secondary"] = 1458] = "Secondary";
    fields[fields["SecondaryCategory"] = 1459] = "SecondaryCategory";
    fields[fields["SecondChanceEligible"] = 1460] = "SecondChanceEligible";
    fields[fields["SecondChanceEnabled"] = 1461] = "SecondChanceEnabled";
    fields[fields["SecondChanceOffer"] = 1462] = "SecondChanceOffer";
    fields[fields["SecondChanceOfferCondition"] = 1463] = "SecondChanceOfferCondition";
    fields[fields["SecondChanceOfferCount"] = 1464] = "SecondChanceOfferCount";
    fields[fields["SecondChanceOfferSent"] = 1465] = "SecondChanceOfferSent";
    fields[fields["SecondChanceOriginalItemID"] = 1466] = "SecondChanceOriginalItemID";
    fields[fields["SecretID"] = 1467] = "SecretID";
    fields[fields["SelectionMode"] = 1468] = "SelectionMode";
    fields[fields["Seller"] = 1469] = "Seller";
    fields[fields["SellerAccount"] = 1470] = "SellerAccount";
    fields[fields["SellerBusinessType"] = 1471] = "SellerBusinessType";
    fields[fields["SellerContactBuyerByEmail"] = 1472] = "SellerContactBuyerByEmail";
    fields[fields["SellerContactDetails"] = 1473] = "SellerContactDetails";
    fields[fields["SellerContactDetailsEnabled"] = 1474] = "SellerContactDetailsEnabled";
    fields[fields["SellerDiscount"] = 1475] = "SellerDiscount";
    fields[fields["SellerDiscounts"] = 1476] = "SellerDiscounts";
    fields[fields["SellereBayPaymentProcessConsent"] = 1477] = "SellereBayPaymentProcessConsent";
    fields[fields["SellereBayPaymentProcessEnableTime"] = 1478] = "SellereBayPaymentProcessEnableTime";
    fields[fields["SellereBayPaymentProcessStatus"] = 1479] = "SellereBayPaymentProcessStatus";
    fields[fields["SellerEIASToken"] = 1480] = "SellerEIASToken";
    fields[fields["SellerEmail"] = 1481] = "SellerEmail";
    fields[fields["SellerExcludeShipToLocationPreferences"] = 1482] = "SellerExcludeShipToLocationPreferences";
    fields[fields["SellerExcludeShipToLocationsPreference"] = 1483] = "SellerExcludeShipToLocationsPreference";
    fields[fields["SellerExternalTransactionID"] = 1484] = "SellerExternalTransactionID";
    fields[fields["SellerFavoriteItemPreferences"] = 1485] = "SellerFavoriteItemPreferences";
    fields[fields["SellerFeeDiscount"] = 1486] = "SellerFeeDiscount";
    fields[fields["SellerGuaranteeEligible"] = 1487] = "SellerGuaranteeEligible";
    fields[fields["SellerGuaranteeLevel"] = 1488] = "SellerGuaranteeLevel";
    fields[fields["SellerID"] = 1489] = "SellerID";
    fields[fields["SellerIDExclude"] = 1490] = "SellerIDExclude";
    fields[fields["SellerInfo"] = 1491] = "SellerInfo";
    fields[fields["SellerInventoryID"] = 1492] = "SellerInventoryID";
    fields[fields["SellerInvoiceNumber"] = 1493] = "SellerInvoiceNumber";
    fields[fields["SellerInvoiceTime"] = 1494] = "SellerInvoiceTime";
    fields[fields["SellerItemRatingDetailArray"] = 1495] = "SellerItemRatingDetailArray";
    fields[fields["SellerLevel"] = 1496] = "SellerLevel";
    fields[fields["SellerMessage"] = 1497] = "SellerMessage";
    fields[fields["SellerNoteToBuyer"] = 1498] = "SellerNoteToBuyer";
    fields[fields["SellerPaidStatus"] = 1499] = "SellerPaidStatus";
    fields[fields["SellerPayment"] = 1500] = "SellerPayment";
    fields[fields["SellerPaymentAddress"] = 1501] = "SellerPaymentAddress";
    fields[fields["SellerPaymentPreferences"] = 1502] = "SellerPaymentPreferences";
    fields[fields["SellerPaymentProfile"] = 1503] = "SellerPaymentProfile";
    fields[fields["SellerProfileOptedIn"] = 1504] = "SellerProfileOptedIn";
    fields[fields["SellerProfilePreferences"] = 1505] = "SellerProfilePreferences";
    fields[fields["SellerProfiles"] = 1506] = "SellerProfiles";
    fields[fields["SellerProvidedTitle"] = 1507] = "SellerProvidedTitle";
    fields[fields["SellerProvidedTitleSupported"] = 1508] = "SellerProvidedTitleSupported";
    fields[fields["SellerRatingSummaryArray"] = 1509] = "SellerRatingSummaryArray";
    fields[fields["SellerResponse"] = 1510] = "SellerResponse";
    fields[fields["SellerReturnPreferences"] = 1511] = "SellerReturnPreferences";
    fields[fields["SellerReturnProfile"] = 1512] = "SellerReturnProfile";
    fields[fields["SellerRoleMetrics"] = 1513] = "SellerRoleMetrics";
    fields[fields["SellerShipmentToLogisticsProvider"] = 1514] = "SellerShipmentToLogisticsProvider";
    fields[fields["SellerShippingProfile"] = 1515] = "SellerShippingProfile";
    fields[fields["SellerThirdPartyCheckoutDisabled"] = 1516] = "SellerThirdPartyCheckoutDisabled";
    fields[fields["SellerThirdPartyUsername"] = 1517] = "SellerThirdPartyUsername";
    fields[fields["SellerUserID"] = 1518] = "SellerUserID";
    fields[fields["SellerVacationNote"] = 1519] = "SellerVacationNote";
    fields[fields["SellingManagerFolderDetails"] = 1520] = "SellingManagerFolderDetails";
    fields[fields["SellingManagerProduct"] = 1521] = "SellingManagerProduct";
    fields[fields["SellingManagerProductDetails"] = 1522] = "SellingManagerProductDetails";
    fields[fields["SellingManagerProductInventoryStatus"] = 1523] = "SellingManagerProductInventoryStatus";
    fields[fields["SellingManagerProductSpecifics"] = 1524] = "SellingManagerProductSpecifics";
    fields[fields["SellingManagerSalesRecordNumber"] = 1525] = "SellingManagerSalesRecordNumber";
    fields[fields["SellingManagerSoldOrder"] = 1526] = "SellingManagerSoldOrder";
    fields[fields["SellingManagerSoldTransaction"] = 1527] = "SellingManagerSoldTransaction";
    fields[fields["SellingManagerTemplateDetails"] = 1528] = "SellingManagerTemplateDetails";
    fields[fields["SellingManagerTemplateDetailsArray"] = 1529] = "SellingManagerTemplateDetailsArray";
    fields[fields["SellingReminders"] = 1530] = "SellingReminders";
    fields[fields["SellingStatus"] = 1531] = "SellingStatus";
    fields[fields["SellingSummary"] = 1532] = "SellingSummary";
    fields[fields["Sender"] = 1533] = "Sender";
    fields[fields["SenderEmail"] = 1534] = "SenderEmail";
    fields[fields["SenderID"] = 1535] = "SenderID";
    fields[fields["SendToName"] = 1536] = "SendToName";
    fields[fields["ServiceType"] = 1537] = "ServiceType";
    fields[fields["SessionID"] = 1538] = "SessionID";
    fields[fields["Severity"] = 1539] = "Severity";
    fields[fields["SeverityCode"] = 1540] = "SeverityCode";
    fields[fields["Shipment"] = 1541] = "Shipment";
    fields[fields["ShipmentLineItem"] = 1542] = "ShipmentLineItem";
    fields[fields["ShipmentTrackingDetails"] = 1543] = "ShipmentTrackingDetails";
    fields[fields["ShipmentTrackingNumber"] = 1544] = "ShipmentTrackingNumber";
    fields[fields["ShipmentTrackNumber"] = 1545] = "ShipmentTrackNumber";
    fields[fields["Shipped"] = 1546] = "Shipped";
    fields[fields["ShippedStatus"] = 1547] = "ShippedStatus";
    fields[fields["ShippedTime"] = 1548] = "ShippedTime";
    fields[fields["ShippingAddress"] = 1549] = "ShippingAddress";
    fields[fields["ShippingCarrier"] = 1550] = "ShippingCarrier";
    fields[fields["ShippingCarrierDetails"] = 1551] = "ShippingCarrierDetails";
    fields[fields["ShippingCarrierID"] = 1552] = "ShippingCarrierID";
    fields[fields["ShippingCarrierUsed"] = 1553] = "ShippingCarrierUsed";
    fields[fields["ShippingCategory"] = 1554] = "ShippingCategory";
    fields[fields["ShippingCategoryDetails"] = 1555] = "ShippingCategoryDetails";
    fields[fields["ShippingConvenienceCharge"] = 1556] = "ShippingConvenienceCharge";
    fields[fields["ShippingCost"] = 1557] = "ShippingCost";
    fields[fields["ShippingCostPaidBy"] = 1558] = "ShippingCostPaidBy";
    fields[fields["ShippingCostPaidByOption"] = 1559] = "ShippingCostPaidByOption";
    fields[fields["ShippingDetails"] = 1560] = "ShippingDetails";
    fields[fields["ShippingDetailsToBeProvidedCount"] = 1561] = "ShippingDetailsToBeProvidedCount";
    fields[fields["ShippingDiscount"] = 1562] = "ShippingDiscount";
    fields[fields["ShippingDiscountAmount"] = 1563] = "ShippingDiscountAmount";
    fields[fields["ShippingDiscountProfileID"] = 1564] = "ShippingDiscountProfileID";
    fields[fields["ShippingIncludedInTax"] = 1565] = "ShippingIncludedInTax";
    fields[fields["ShippingInsurance"] = 1566] = "ShippingInsurance";
    fields[fields["ShippingInsuranceCost"] = 1567] = "ShippingInsuranceCost";
    fields[fields["ShippingIrregular"] = 1568] = "ShippingIrregular";
    fields[fields["ShippingLocation"] = 1569] = "ShippingLocation";
    fields[fields["ShippingLocationDetails"] = 1570] = "ShippingLocationDetails";
    fields[fields["ShippingNeededCount"] = 1571] = "ShippingNeededCount";
    fields[fields["ShippingPackage"] = 1572] = "ShippingPackage";
    fields[fields["ShippingPackageDetails"] = 1573] = "ShippingPackageDetails";
    fields[fields["ShippingPackageInfo"] = 1574] = "ShippingPackageInfo";
    fields[fields["ShippingProfileCategoryGroup"] = 1575] = "ShippingProfileCategoryGroup";
    fields[fields["ShippingProfileID"] = 1576] = "ShippingProfileID";
    fields[fields["ShippingProfileName"] = 1577] = "ShippingProfileName";
    fields[fields["ShippingRateErrorMessage"] = 1578] = "ShippingRateErrorMessage";
    fields[fields["ShippingRateType"] = 1579] = "ShippingRateType";
    fields[fields["ShippingReimbursement"] = 1580] = "ShippingReimbursement";
    fields[fields["ShippingService"] = 1581] = "ShippingService";
    fields[fields["ShippingServiceAdditionalCost"] = 1582] = "ShippingServiceAdditionalCost";
    fields[fields["ShippingServiceCost"] = 1583] = "ShippingServiceCost";
    fields[fields["ShippingServiceCostOverride"] = 1584] = "ShippingServiceCostOverride";
    fields[fields["ShippingServiceCostOverrideList"] = 1585] = "ShippingServiceCostOverrideList";
    fields[fields["ShippingServiceCutOffTime"] = 1586] = "ShippingServiceCutOffTime";
    fields[fields["ShippingServiceDetails"] = 1587] = "ShippingServiceDetails";
    fields[fields["ShippingServiceID"] = 1588] = "ShippingServiceID";
    fields[fields["ShippingServiceOptions"] = 1589] = "ShippingServiceOptions";
    fields[fields["ShippingServicePackageDetails"] = 1590] = "ShippingServicePackageDetails";
    fields[fields["ShippingServicePriority"] = 1591] = "ShippingServicePriority";
    fields[fields["ShippingServiceSelected"] = 1592] = "ShippingServiceSelected";
    fields[fields["ShippingServiceType"] = 1593] = "ShippingServiceType";
    fields[fields["ShippingServiceUsed"] = 1594] = "ShippingServiceUsed";
    fields[fields["ShippingSurcharge"] = 1595] = "ShippingSurcharge";
    fields[fields["ShippingTermsInDescription"] = 1596] = "ShippingTermsInDescription";
    fields[fields["ShippingTermsRequired"] = 1597] = "ShippingTermsRequired";
    fields[fields["ShippingTime"] = 1598] = "ShippingTime";
    fields[fields["ShippingTimeMax"] = 1599] = "ShippingTimeMax";
    fields[fields["ShippingTimeMin"] = 1600] = "ShippingTimeMin";
    fields[fields["ShippingTrackingEvent"] = 1601] = "ShippingTrackingEvent";
    fields[fields["ShippingType"] = 1602] = "ShippingType";
    fields[fields["ShipToAddress"] = 1603] = "ShipToAddress";
    fields[fields["ShipToLocation"] = 1604] = "ShipToLocation";
    fields[fields["ShipToLocations"] = 1605] = "ShipToLocations";
    fields[fields["ShipToRegistrationCountry"] = 1606] = "ShipToRegistrationCountry";
    fields[fields["ShortMessage"] = 1607] = "ShortMessage";
    fields[fields["ShortSummary"] = 1608] = "ShortSummary";
    fields[fields["ShowBidderNoticePreferences"] = 1609] = "ShowBidderNoticePreferences";
    fields[fields["ShowCombinedPaymentPreferences"] = 1610] = "ShowCombinedPaymentPreferences";
    fields[fields["ShowCrossPromotionPreferences"] = 1611] = "ShowCrossPromotionPreferences";
    fields[fields["ShowDispatchCutoffTimePreferences"] = 1612] = "ShowDispatchCutoffTimePreferences";
    fields[fields["ShowEmailShipmentTrackingNumberPreference"] = 1613] = "ShowEmailShipmentTrackingNumberPreference";
    fields[fields["ShowEndOfAuctionEmailPreferences"] = 1614] = "ShowEndOfAuctionEmailPreferences";
    fields[fields["ShowGlobalShippingProgramListingPreference"] = 1615] = "ShowGlobalShippingProgramListingPreference";
    fields[fields["ShowGlobalShippingProgramPreference"] = 1616] = "ShowGlobalShippingProgramPreference";
    fields[fields["ShowOverrideGSPServiceWithIntlServicePreference"] = 1617] = "ShowOverrideGSPServiceWithIntlServicePreference";
    fields[fields["ShowProStoresPreferences"] = 1618] = "ShowProStoresPreferences";
    fields[fields["ShowPurchaseReminderEmailPreferences"] = 1619] = "ShowPurchaseReminderEmailPreferences";
    fields[fields["ShowRequiredShipPhoneNumberPreference"] = 1620] = "ShowRequiredShipPhoneNumberPreference";
    fields[fields["ShowSellerExcludeShipToLocationPreference"] = 1621] = "ShowSellerExcludeShipToLocationPreference";
    fields[fields["ShowSellerFavoriteItemPreferences"] = 1622] = "ShowSellerFavoriteItemPreferences";
    fields[fields["ShowSellerPaymentPreferences"] = 1623] = "ShowSellerPaymentPreferences";
    fields[fields["ShowSellerProfilePreferences"] = 1624] = "ShowSellerProfilePreferences";
    fields[fields["ShowSellerReturnPreferences"] = 1625] = "ShowSellerReturnPreferences";
    fields[fields["ShowUnpaidItemAssistanceExclusionList"] = 1626] = "ShowUnpaidItemAssistanceExclusionList";
    fields[fields["ShowUnpaidItemAssistancePreference"] = 1627] = "ShowUnpaidItemAssistancePreference";
    fields[fields["SignUpForStoreNewsletter"] = 1628] = "SignUpForStoreNewsletter";
    fields[fields["Site"] = 1629] = "Site";
    fields[fields["SiteCurrency"] = 1630] = "SiteCurrency";
    fields[fields["SiteDefaults"] = 1631] = "SiteDefaults";
    fields[fields["SiteDetails"] = 1632] = "SiteDetails";
    fields[fields["SiteHostedPictureDetails"] = 1633] = "SiteHostedPictureDetails";
    fields[fields["SiteID"] = 1634] = "SiteID";
    fields[fields["SiteWideCharacteristicSets"] = 1635] = "SiteWideCharacteristicSets";
    fields[fields["SKU"] = 1636] = "SKU";
    fields[fields["SKUArray"] = 1637] = "SKUArray";
    fields[fields["SkypeContactOption"] = 1638] = "SkypeContactOption";
    fields[fields["SkypeEnabled"] = 1639] = "SkypeEnabled";
    fields[fields["SkypeID"] = 1640] = "SkypeID";
    fields[fields["SkypeMeNonTransactionalEnabled"] = 1641] = "SkypeMeNonTransactionalEnabled";
    fields[fields["SkypeMeTransactionalEnabled"] = 1642] = "SkypeMeTransactionalEnabled";
    fields[fields["SMSPhone"] = 1643] = "SMSPhone";
    fields[fields["SMSSubscription"] = 1644] = "SMSSubscription";
    fields[fields["SoldAlert"] = 1645] = "SoldAlert";
    fields[fields["SoldAsBin"] = 1646] = "SoldAsBin";
    fields[fields["SoldDurationInDays"] = 1647] = "SoldDurationInDays";
    fields[fields["SoldList"] = 1648] = "SoldList";
    fields[fields["SoldOffeBay"] = 1649] = "SoldOffeBay";
    fields[fields["SoldOn"] = 1650] = "SoldOn";
    fields[fields["SoldOneBay"] = 1651] = "SoldOneBay";
    fields[fields["Sort"] = 1652] = "Sort";
    fields[fields["SortingOrder"] = 1653] = "SortingOrder";
    fields[fields["SortOrder"] = 1654] = "SortOrder";
    fields[fields["Source"] = 1655] = "Source";
    fields[fields["SourceSaleTemplateID"] = 1656] = "SourceSaleTemplateID";
    fields[fields["SpacingIntervalInMinutes"] = 1657] = "SpacingIntervalInMinutes";
    fields[fields["SpecialitySubscription"] = 1658] = "SpecialitySubscription";
    fields[fields["StandardLabel"] = 1659] = "StandardLabel";
    fields[fields["StandardOffset"] = 1660] = "StandardOffset";
    fields[fields["StartCreationTime"] = 1661] = "StartCreationTime";
    fields[fields["StartingDisputeID"] = 1662] = "StartingDisputeID";
    fields[fields["StartPrice"] = 1663] = "StartPrice";
    fields[fields["StartTime"] = 1664] = "StartTime";
    fields[fields["StartTimeFrom"] = 1665] = "StartTimeFrom";
    fields[fields["StartTimeTo"] = 1666] = "StartTimeTo";
    fields[fields["StateOrProvince"] = 1667] = "StateOrProvince";
    fields[fields["StaticAlias"] = 1668] = "StaticAlias";
    fields[fields["Status"] = 1669] = "Status";
    fields[fields["StockPhoto"] = 1670] = "StockPhoto";
    fields[fields["StockPhotoURL"] = 1671] = "StockPhotoURL";
    fields[fields["Store"] = 1672] = "Store";
    fields[fields["StoreCategories"] = 1673] = "StoreCategories";
    fields[fields["StoreCategory2ID"] = 1674] = "StoreCategory2ID";
    fields[fields["StoreCategory2Name"] = 1675] = "StoreCategory2Name";
    fields[fields["StoreCategoryID"] = 1676] = "StoreCategoryID";
    fields[fields["StoreCategoryName"] = 1677] = "StoreCategoryName";
    fields[fields["StoredComments"] = 1678] = "StoredComments";
    fields[fields["StoredCommentText"] = 1679] = "StoredCommentText";
    fields[fields["Storefront"] = 1680] = "Storefront";
    fields[fields["StoreID"] = 1681] = "StoreID";
    fields[fields["StoreInventoryEnabled"] = 1682] = "StoreInventoryEnabled";
    fields[fields["StoreName"] = 1683] = "StoreName";
    fields[fields["StoreOwner"] = 1684] = "StoreOwner";
    fields[fields["StoreOwnerExtendedListingDurations"] = 1685] = "StoreOwnerExtendedListingDurations";
    fields[fields["StoreOwnerExtendedListingDurationsEnabled"] = 1686] = "StoreOwnerExtendedListingDurationsEnabled";
    fields[fields["StorePreferences"] = 1687] = "StorePreferences";
    fields[fields["StoreSite"] = 1688] = "StoreSite";
    fields[fields["StoreURL"] = 1689] = "StoreURL";
    fields[fields["Street"] = 1690] = "Street";
    fields[fields["Street1"] = 1691] = "Street1";
    fields[fields["Street2"] = 1692] = "Street2";
    fields[fields["Subject"] = 1693] = "Subject";
    fields[fields["SubmittedTime"] = 1694] = "SubmittedTime";
    fields[fields["SubscribeSA"] = 1695] = "SubscribeSA";
    fields[fields["SubscribeSAPro"] = 1696] = "SubscribeSAPro";
    fields[fields["SubscribeSM"] = 1697] = "SubscribeSM";
    fields[fields["SubscribeSMPro"] = 1698] = "SubscribeSMPro";
    fields[fields["Subscription"] = 1699] = "Subscription";
    fields[fields["SubscriptionArray"] = 1700] = "SubscriptionArray";
    fields[fields["SubscriptionLevel"] = 1701] = "SubscriptionLevel";
    fields[fields["SubTitle"] = 1702] = "SubTitle";
    fields[fields["Subtotal"] = 1703] = "Subtotal";
    fields[fields["SubtotalAmount"] = 1704] = "SubtotalAmount";
    fields[fields["SuccessPercent"] = 1705] = "SuccessPercent";
    fields[fields["SuggestedCategory"] = 1706] = "SuggestedCategory";
    fields[fields["SuggestedCategoryArray"] = 1707] = "SuggestedCategoryArray";
    fields[fields["SuggestedText"] = 1708] = "SuggestedText";
    fields[fields["Summary"] = 1709] = "Summary";
    fields[fields["SummaryDays"] = 1710] = "SummaryDays";
    fields[fields["SummaryPeriod"] = 1711] = "SummaryPeriod";
    fields[fields["SummarySchedule"] = 1712] = "SummarySchedule";
    fields[fields["SupportedSellerProfile"] = 1713] = "SupportedSellerProfile";
    fields[fields["SupportedSellerProfiles"] = 1714] = "SupportedSellerProfiles";
    fields[fields["SupportedSite"] = 1715] = "SupportedSite";
    fields[fields["SurchargeApplicable"] = 1716] = "SurchargeApplicable";
    fields[fields["TargetUser"] = 1717] = "TargetUser";
    fields[fields["TargetUserID"] = 1718] = "TargetUserID";
    fields[fields["TaskID"] = 1719] = "TaskID";
    fields[fields["TaskReferenceID"] = 1720] = "TaskReferenceID";
    fields[fields["TaxAmount"] = 1721] = "TaxAmount";
    fields[fields["TaxCategory"] = 1722] = "TaxCategory";
    fields[fields["TaxDescription"] = 1723] = "TaxDescription";
    fields[fields["TaxDetails"] = 1724] = "TaxDetails";
    fields[fields["Taxes"] = 1725] = "Taxes";
    fields[fields["TaxJurisdiction"] = 1726] = "TaxJurisdiction";
    fields[fields["TaxOnHandlingAmount"] = 1727] = "TaxOnHandlingAmount";
    fields[fields["TaxOnShippingAmount"] = 1728] = "TaxOnShippingAmount";
    fields[fields["TaxOnSubtotalAmount"] = 1729] = "TaxOnSubtotalAmount";
    fields[fields["TaxTable"] = 1730] = "TaxTable";
    fields[fields["TCROriginalItemID"] = 1731] = "TCROriginalItemID";
    fields[fields["Template"] = 1732] = "Template";
    fields[fields["TemplateID"] = 1733] = "TemplateID";
    fields[fields["TemplateName"] = 1734] = "TemplateName";
    fields[fields["TemplateText"] = 1735] = "TemplateText";
    fields[fields["TemplateXML"] = 1736] = "TemplateXML";
    fields[fields["TermsAndConditions"] = 1737] = "TermsAndConditions";
    fields[fields["Text"] = 1738] = "Text";
    fields[fields["TextCustomized"] = 1739] = "TextCustomized";
    fields[fields["Theme"] = 1740] = "Theme";
    fields[fields["ThemeGroup"] = 1741] = "ThemeGroup";
    fields[fields["ThemeID"] = 1742] = "ThemeID";
    fields[fields["ThemeTotal"] = 1743] = "ThemeTotal";
    fields[fields["ThirdPartyCheckout"] = 1744] = "ThirdPartyCheckout";
    fields[fields["TicketListingDetails"] = 1745] = "TicketListingDetails";
    fields[fields["Time"] = 1746] = "Time";
    fields[fields["TimeBid"] = 1747] = "TimeBid";
    fields[fields["TimeFrom"] = 1748] = "TimeFrom";
    fields[fields["TimeLeft"] = 1749] = "TimeLeft";
    fields[fields["Timestamp"] = 1750] = "Timestamp";
    fields[fields["TimeTo"] = 1751] = "TimeTo";
    fields[fields["TimeZoneDetails"] = 1752] = "TimeZoneDetails";
    fields[fields["TimeZoneID"] = 1753] = "TimeZoneID";
    fields[fields["Title"] = 1754] = "Title";
    fields[fields["TitleColor"] = 1755] = "TitleColor";
    fields[fields["TitleFace"] = 1756] = "TitleFace";
    fields[fields["TitleSize"] = 1757] = "TitleSize";
    fields[fields["TitleWithCompatibility"] = 1758] = "TitleWithCompatibility";
    fields[fields["TokenStatus"] = 1759] = "TokenStatus";
    fields[fields["TopRatedListing"] = 1760] = "TopRatedListing";
    fields[fields["TopRatedProgram"] = 1761] = "TopRatedProgram";
    fields[fields["TopRatedSeller"] = 1762] = "TopRatedSeller";
    fields[fields["TopRatedSellerDetails"] = 1763] = "TopRatedSellerDetails";
    fields[fields["Total"] = 1764] = "Total";
    fields[fields["TotalAlertCount"] = 1765] = "TotalAlertCount";
    fields[fields["TotalAmount"] = 1766] = "TotalAmount";
    fields[fields["TotalAuctionSellingValue"] = 1767] = "TotalAuctionSellingValue";
    fields[fields["TotalAvailable"] = 1768] = "TotalAvailable";
    fields[fields["TotalBids"] = 1769] = "TotalBids";
    fields[fields["TotalEmailsSent"] = 1770] = "TotalEmailsSent";
    fields[fields["TotalFeedbackPeriodArray"] = 1771] = "TotalFeedbackPeriodArray";
    fields[fields["TotalHighPriorityCount"] = 1772] = "TotalHighPriorityCount";
    fields[fields["TotalLeadCount"] = 1773] = "TotalLeadCount";
    fields[fields["TotalListingsWithLeads"] = 1774] = "TotalListingsWithLeads";
    fields[fields["TotalMessageCount"] = 1775] = "TotalMessageCount";
    fields[fields["TotalNewLeadsCount"] = 1776] = "TotalNewLeadsCount";
    fields[fields["TotalNumberOfEntries"] = 1777] = "TotalNumberOfEntries";
    fields[fields["TotalNumberOfPages"] = 1778] = "TotalNumberOfPages";
    fields[fields["TotalPrice"] = 1779] = "TotalPrice";
    fields[fields["TotalQuantity"] = 1780] = "TotalQuantity";
    fields[fields["TotalQuestionCount"] = 1781] = "TotalQuestionCount";
    fields[fields["TotalRefundToBuyer"] = 1782] = "TotalRefundToBuyer";
    fields[fields["TotalShippingCost"] = 1783] = "TotalShippingCost";
    fields[fields["TotalSoldCount"] = 1784] = "TotalSoldCount";
    fields[fields["TotalSoldValue"] = 1785] = "TotalSoldValue";
    fields[fields["TotalTaxAmount"] = 1786] = "TotalTaxAmount";
    fields[fields["TotalWinningCost"] = 1787] = "TotalWinningCost";
    fields[fields["TotalWonCost"] = 1788] = "TotalWonCost";
    fields[fields["TrackingID"] = 1789] = "TrackingID";
    fields[fields["TrackingPartnerCode"] = 1790] = "TrackingPartnerCode";
    fields[fields["TradeInMake"] = 1791] = "TradeInMake";
    fields[fields["TradeInModel"] = 1792] = "TradeInModel";
    fields[fields["TradeInYear"] = 1793] = "TradeInYear";
    fields[fields["TradeRegistrationNumber"] = 1794] = "TradeRegistrationNumber";
    fields[fields["Transaction"] = 1795] = "Transaction";
    fields[fields["TransactionArray"] = 1796] = "TransactionArray";
    fields[fields["TransactionConfirmationRequestEnabled"] = 1797] = "TransactionConfirmationRequestEnabled";
    fields[fields["TransactionID"] = 1798] = "TransactionID";
    fields[fields["TransactionPercent"] = 1799] = "TransactionPercent";
    fields[fields["TransactionPrice"] = 1800] = "TransactionPrice";
    fields[fields["TransactionSiteID"] = 1801] = "TransactionSiteID";
    fields[fields["TransactionsPerPage"] = 1802] = "TransactionsPerPage";
    fields[fields["TUVLevel"] = 1803] = "TUVLevel";
    fields[fields["Type"] = 1804] = "Type";
    fields[fields["UniqueBuyerCount"] = 1805] = "UniqueBuyerCount";
    fields[fields["UniqueNegativeFeedbackCount"] = 1806] = "UniqueNegativeFeedbackCount";
    fields[fields["UniqueNeutralFeedbackCount"] = 1807] = "UniqueNeutralFeedbackCount";
    fields[fields["UniquePositiveFeedbackCount"] = 1808] = "UniquePositiveFeedbackCount";
    fields[fields["UnitCost"] = 1809] = "UnitCost";
    fields[fields["UnitInfo"] = 1810] = "UnitInfo";
    fields[fields["UnitOfMeasurement"] = 1811] = "UnitOfMeasurement";
    fields[fields["UnitOfMeasurementDetails"] = 1812] = "UnitOfMeasurementDetails";
    fields[fields["UnitQuantity"] = 1813] = "UnitQuantity";
    fields[fields["UnitsAvailable"] = 1814] = "UnitsAvailable";
    fields[fields["UnitType"] = 1815] = "UnitType";
    fields[fields["UnmappedCategoryArray"] = 1816] = "UnmappedCategoryArray";
    fields[fields["UnpaidItem"] = 1817] = "UnpaidItem";
    fields[fields["UnpaidItemAssistancePreferences"] = 1818] = "UnpaidItemAssistancePreferences";
    fields[fields["UnpaidItemStatus"] = 1819] = "UnpaidItemStatus";
    fields[fields["UnresolvedAlertCount"] = 1820] = "UnresolvedAlertCount";
    fields[fields["UnsoldList"] = 1821] = "UnsoldList";
    fields[fields["UnsubscribeNotification"] = 1822] = "UnsubscribeNotification";
    fields[fields["UnsuccessfulBidderNoticeIncludeMyItems"] = 1823] = "UnsuccessfulBidderNoticeIncludeMyItems";
    fields[fields["UPC"] = 1824] = "UPC";
    fields[fields["UPCIdentifierEnabled"] = 1825] = "UPCIdentifierEnabled";
    fields[fields["Updated"] = 1826] = "Updated";
    fields[fields["UpdateReturnPolicy"] = 1827] = "UpdateReturnPolicy";
    fields[fields["UpdateSellerInfo"] = 1828] = "UpdateSellerInfo";
    fields[fields["UpdateTime"] = 1829] = "UpdateTime";
    fields[fields["UpSellGallerySortFilter"] = 1830] = "UpSellGallerySortFilter";
    fields[fields["UpSellItemFormatSortFilter"] = 1831] = "UpSellItemFormatSortFilter";
    fields[fields["UpSellItemSortFilter"] = 1832] = "UpSellItemSortFilter";
    fields[fields["UPSRateOption"] = 1833] = "UPSRateOption";
    fields[fields["URL"] = 1834] = "URL";
    fields[fields["URLDetails"] = 1835] = "URLDetails";
    fields[fields["URLPath"] = 1836] = "URLPath";
    fields[fields["URLType"] = 1837] = "URLType";
    fields[fields["UseByDate"] = 1838] = "UseByDate";
    fields[fields["UseFirstProduct"] = 1839] = "UseFirstProduct";
    fields[fields["User"] = 1840] = "User";
    fields[fields["UserAgreementInfo"] = 1841] = "UserAgreementInfo";
    fields[fields["UserAgreementURL"] = 1842] = "UserAgreementURL";
    fields[fields["UserAnonymized"] = 1843] = "UserAnonymized";
    fields[fields["UserConsent"] = 1844] = "UserConsent";
    fields[fields["UserConsentRequired"] = 1845] = "UserConsentRequired";
    fields[fields["UserData"] = 1846] = "UserData";
    fields[fields["UserDefinedList"] = 1847] = "UserDefinedList";
    fields[fields["UserDefinedListName"] = 1848] = "UserDefinedListName";
    fields[fields["UserDefinedLists"] = 1849] = "UserDefinedLists";
    fields[fields["UserDeliveryPreferenceArray"] = 1850] = "UserDeliveryPreferenceArray";
    fields[fields["UserDisplayHint"] = 1851] = "UserDisplayHint";
    fields[fields["UseRecommendedProduct"] = 1852] = "UseRecommendedProduct";
    fields[fields["UserID"] = 1853] = "UserID";
    fields[fields["UserIDChanged"] = 1854] = "UserIDChanged";
    fields[fields["UserIDLastChanged"] = 1855] = "UserIDLastChanged";
    fields[fields["UserInput"] = 1856] = "UserInput";
    fields[fields["UserRole"] = 1857] = "UserRole";
    fields[fields["UserStatus"] = 1858] = "UserStatus";
    fields[fields["UserSubscription"] = 1859] = "UserSubscription";
    fields[fields["UseStockPhotoURLAsGallery"] = 1860] = "UseStockPhotoURLAsGallery";
    fields[fields["UseTaxTable"] = 1861] = "UseTaxTable";
    fields[fields["USPSRateOption"] = 1862] = "USPSRateOption";
    fields[fields["UUID"] = 1863] = "UUID";
    fields[fields["VacationPreferences"] = 1864] = "VacationPreferences";
    fields[fields["ValidationRules"] = 1865] = "ValidationRules";
    fields[fields["ValidForSellingFlow"] = 1866] = "ValidForSellingFlow";
    fields[fields["ValidToken"] = 1867] = "ValidToken";
    fields[fields["Value"] = 1868] = "Value";
    fields[fields["ValueCategory"] = 1869] = "ValueCategory";
    fields[fields["ValueFormat"] = 1870] = "ValueFormat";
    fields[fields["ValueLiteral"] = 1871] = "ValueLiteral";
    fields[fields["ValuePackEnabled"] = 1872] = "ValuePackEnabled";
    fields[fields["ValueRecommendation"] = 1873] = "ValueRecommendation";
    fields[fields["ValueType"] = 1874] = "ValueType";
    fields[fields["Variation"] = 1875] = "Variation";
    fields[fields["VariationDetails"] = 1876] = "VariationDetails";
    fields[fields["VariationKey"] = 1877] = "VariationKey";
    fields[fields["VariationPicture"] = 1878] = "VariationPicture";
    fields[fields["Variations"] = 1879] = "Variations";
    fields[fields["VariationsEnabled"] = 1880] = "VariationsEnabled";
    fields[fields["VariationSKU"] = 1881] = "VariationSKU";
    fields[fields["VariationSpecificName"] = 1882] = "VariationSpecificName";
    fields[fields["VariationSpecificPictureSet"] = 1883] = "VariationSpecificPictureSet";
    fields[fields["VariationSpecifics"] = 1884] = "VariationSpecifics";
    fields[fields["VariationSpecificsSet"] = 1885] = "VariationSpecificsSet";
    fields[fields["VariationSpecificValue"] = 1886] = "VariationSpecificValue";
    fields[fields["VariationTitle"] = 1887] = "VariationTitle";
    fields[fields["VariationViewItemURL"] = 1888] = "VariationViewItemURL";
    fields[fields["VATDetails"] = 1889] = "VATDetails";
    fields[fields["VATID"] = 1890] = "VATID";
    fields[fields["VATInsuranceFee"] = 1891] = "VATInsuranceFee";
    fields[fields["VATPercent"] = 1892] = "VATPercent";
    fields[fields["VATRate"] = 1893] = "VATRate";
    fields[fields["VATShippingFee"] = 1894] = "VATShippingFee";
    fields[fields["VATSite"] = 1895] = "VATSite";
    fields[fields["VATStatus"] = 1896] = "VATStatus";
    fields[fields["VATTotalAmount"] = 1897] = "VATTotalAmount";
    fields[fields["VendorContactInfo"] = 1898] = "VendorContactInfo";
    fields[fields["VendorInfo"] = 1899] = "VendorInfo";
    fields[fields["VendorName"] = 1900] = "VendorName";
    fields[fields["Venue"] = 1901] = "Venue";
    fields[fields["VerifiedUser"] = 1902] = "VerifiedUser";
    fields[fields["VerifiedUserRequirements"] = 1903] = "VerifiedUserRequirements";
    fields[fields["VerifyEligibilityOnly"] = 1904] = "VerifyEligibilityOnly";
    fields[fields["VerifyOnly"] = 1905] = "VerifyOnly";
    fields[fields["VeROReasonCodeDetails"] = 1906] = "VeROReasonCodeDetails";
    fields[fields["VeROReasonCodeID"] = 1907] = "VeROReasonCodeID";
    fields[fields["VeROReportPacketID"] = 1908] = "VeROReportPacketID";
    fields[fields["VeROReportPacketStatus"] = 1909] = "VeROReportPacketStatus";
    fields[fields["VeROSiteDetail"] = 1910] = "VeROSiteDetail";
    fields[fields["Version"] = 1911] = "Version";
    fields[fields["ViewAllNodes"] = 1912] = "ViewAllNodes";
    fields[fields["ViewItemURL"] = 1913] = "ViewItemURL";
    fields[fields["ViewItemURLForNaturalSearch"] = 1914] = "ViewItemURLForNaturalSearch";
    fields[fields["VIN"] = 1915] = "VIN";
    fields[fields["VINLink"] = 1916] = "VINLink";
    fields[fields["VINSupported"] = 1917] = "VINSupported";
    fields[fields["Virtual"] = 1918] = "Virtual";
    fields[fields["VRM"] = 1919] = "VRM";
    fields[fields["VRMLink"] = 1920] = "VRMLink";
    fields[fields["VRMSupported"] = 1921] = "VRMSupported";
    fields[fields["WarningLevel"] = 1922] = "WarningLevel";
    fields[fields["WarrantyDuration"] = 1923] = "WarrantyDuration";
    fields[fields["WarrantyDurationOption"] = 1924] = "WarrantyDurationOption";
    fields[fields["WarrantyOffered"] = 1925] = "WarrantyOffered";
    fields[fields["WarrantyOfferedOption"] = 1926] = "WarrantyOfferedOption";
    fields[fields["WarrantyType"] = 1927] = "WarrantyType";
    fields[fields["WarrantyTypeOption"] = 1928] = "WarrantyTypeOption";
    fields[fields["WatchCount"] = 1929] = "WatchCount";
    fields[fields["WatchList"] = 1930] = "WatchList";
    fields[fields["WatchListCount"] = 1931] = "WatchListCount";
    fields[fields["WatchListMaximum"] = 1932] = "WatchListMaximum";
    fields[fields["WeightMajor"] = 1933] = "WeightMajor";
    fields[fields["WeightMinor"] = 1934] = "WeightMinor";
    fields[fields["WeightOff"] = 1935] = "WeightOff";
    fields[fields["WeightRequired"] = 1936] = "WeightRequired";
    fields[fields["WinningCount"] = 1937] = "WinningCount";
    fields[fields["WonCount"] = 1938] = "WonCount";
    fields[fields["WonDurationInDays"] = 1939] = "WonDurationInDays";
    fields[fields["WonList"] = 1940] = "WonList";
    fields[fields["ZeroFeedbackScore"] = 1941] = "ZeroFeedbackScore";
})(fields || (fields = {}));
exports.default = fields;
